<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marc Noguera-Julian, PhD. TreeTopUnder" />

<meta name="date" content="2022-11-01" />

<title>Metagenomic Analysis of Canine gut microbiome in Leishmaniasis</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Purina_N1H_2022</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="01_index.html">Home</a>
</li>
<li>
  <a href="02_Exploratory.html">Exploratory</a>
</li>
<li>
  <a href="03_Goal1.html">Goal 1</a>
</li>
<li>
  <a href="04_Goal2.html">Goal 2</a>
</li>
<li>
  <a href="05_Goal3.html">Goal 3</a>
</li>
<li>
  <a href="06_Conclusions.html">Conclusions</a>
</li>
<li>
  <a href="MetadataExplorer.html">Metadata</a>
</li>
<li>
  <a href="Multiqc_report.html">MultiQC</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Metagenomic Analysis of Canine gut microbiome in Leishmaniasis</h1>
<h4 class="author">Marc Noguera-Julian, PhD. TreeTopUnder</h4>
<h4 class="date">2022-11-01</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-01
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Purina_N1H_2022/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220906code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220906)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220906code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220906)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTreeTopUnderPurinaN1H2022tree74b0908090ff8e0bc963c89a8444a189c23e53datargetblank74b0908a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/TreeTopUnder/Purina_N1H_2022/tree/74b0908090ff8e0bc963c89a8444a189c23e53da" target="_blank">74b0908</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomTreeTopUnderPurinaN1H2022tree74b0908090ff8e0bc963c89a8444a189c23e53datargetblank74b0908a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/TreeTopUnder/Purina_N1H_2022/tree/74b0908090ff8e0bc963c89a8444a189c23e53da" target="_blank">74b0908</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/001_metadata_check.Rmd
    Ignored:    code/003_metadata_vars_selection.Rmd
    Ignored:    code/006_DataPreprocessing.Rmd
    Ignored:    code/006_DataPreprocessing.html
    Ignored:    code/008_PopulateMRE.Rmd
    Ignored:    code/Phylosmith_bug.Rmd
    Ignored:    code/output/
    Ignored:    data/.DS_Store
    Ignored:    data/generichness_goal3.rds
    Ignored:    data/mreObject_feces.rds
    Ignored:    data/mreObject_goal1.rds
    Ignored:    data/mreObject_goal2.rds
    Ignored:    data/mreObject_goal3.rds
    Ignored:    data/processed/
    Ignored:    data/ps_metaphlan_feces.rds
    Ignored:    data/raw/
    Ignored:    output/.DS_Store
    Ignored:    output/Deliverables/.DS_Store
    Ignored:    output/Deliverables/Exploratory/.DS_Store
    Ignored:    output/Deliverables/Exploratory/DifferentialAbundance/
    Ignored:    output/Deliverables/Goal1/.DS_Store
    Ignored:    output/Deliverables/Goal1/DifferentialAbundance/.DS_Store
    Ignored:    output/Deliverables/Goal1/DifferentialAbundance/lefse/.DS_Store
    Ignored:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/.DS_Store
    Ignored:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/genus/figures/
    Ignored:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/species/figures/
    Ignored:    output/Deliverables/Goal2/.DS_Store
    Ignored:    output/Deliverables/Goal2/DifferentialAbundance/.DS_Store
    Ignored:    output/Deliverables/Goal2/DifferentialAbundance/lefse/.DS_Store
    Ignored:    output/Deliverables/Goal2/DifferentialAbundance/lefse/boxplots/.DS_Store
    Ignored:    output/mre_data/.DS_Store
    Ignored:    output/mre_data/Goal3/
    Ignored:    output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/
    Ignored:    output/mre_data/feces/

Untracked files:
    Untracked:  MetadataExplorer.html
    Untracked:  analysis/06_Conclusions.Rmd
    Untracked:  output/Deliverables/Exploratory/BetaDiversity/Ordination_BLSamples.pdf
    Untracked:  output/Deliverables/Exploratory/DataExport/
    Untracked:  output/Deliverables/Exploratory/StackedBarplots_Genus.pdf
    Untracked:  output/Deliverables/Exploratory/TwoSamples_Efaecium.pdf
    Untracked:  output/Deliverables/ExploratoryHeatmap_allSamples.pdf
    Untracked:  output/Deliverables/Goal1/BetaDiversity/NMDS_Goal1_StressPlot.pdf
    Untracked:  output/Deliverables/Goal1/DifferentialAbundance/maaslin/MaasLin2_M1_vs_BL_function.xls
    Untracked:  output/Deliverables/Goal1/DifferentialAbundance/maaslin/MaasLin2_M1_vs_BL_function.xlsx
    Untracked:  output/Deliverables/Goal1/DifferentialAbundance/maaslin/MaasLin2_M1_vs_BL_function_kegg.pdf
    Untracked:  output/Deliverables/Goal1/DifferentialAbundance/maaslin/MaasLin2_M1_vs_BL_function_kegg.rds
    Untracked:  output/Deliverables/Goal1/DifferentialAbundance/maaslin/MaasLin2_M1_vs_BL_function_kegg.xls
    Untracked:  output/Deliverables/Goal1/DifferentialAbundance/maaslin/MaasLin2_M1_vs_BL_function_metacyc.pdf
    Untracked:  output/Deliverables/Goal1/DifferentialAbundance/maaslin/MaasLin2_M1_vs_BL_function_metacyc.rds
    Untracked:  output/Deliverables/Goal1/DifferentialAbundance/maaslin/MaasLin2_M1_vs_BL_function_metacyc.xls
    Untracked:  output/Deliverables/Goal1/DifferentialAbundance/maaslin/MaasLin2_M1_vs_BL_taxonomy_genus.rds
    Untracked:  output/Deliverables/Goal1/DifferentialAbundance/maaslin/MaasLin2_M1_vs_BL_taxonomy_species.rds
    Untracked:  output/Deliverables/Goal1/DifferentialAbundance/maaslin/MetaCyc/
    Untracked:  output/Deliverables/Goal1/DifferentialAbundance/maaslin/Volcano_M1_vs_BL.xlsx
    Untracked:  output/Deliverables/Goal1/DifferentialAbundance/maaslin/Volcano_M1_vs_BL_taxonomy_genus.pdf
    Untracked:  output/Deliverables/Goal1/DifferentialAbundance/maaslin/Volcano_M1_vs_BL_taxonomy_species.pdf
    Untracked:  output/Deliverables/Goal1/DifferentialAbundance/maaslin/kegg/
    Untracked:  output/Deliverables/Goal2/AlphaDiversity/AlphaDiversity_Timepoint_goal2_Probiotic_samples.pdf
    Untracked:  output/Deliverables/Goal2/DifferentialAbundance/lefse/boxplots/Blautia_sp_CAG_257_vs_Timepoint_Probiotic_Goal2.pdf
    Untracked:  output/Deliverables/Goal2/DifferentialAbundance/lefse/boxplots/Clostridium_sp_CAG_58_vs_Timepoint_Probiotic_Goal2.pdf
    Untracked:  output/Deliverables/Goal2/DifferentialAbundance/lefse/boxplots/Dorea_sp_CAG_317_vs_Timepoint_Probiotic_Goal2.pdf
    Untracked:  output/Deliverables/Goal2/DifferentialAbundance/lefse/boxplots/paired_Blautia_sp_CAG_257_vs_Timepoint_Probiotic_Goal2.pdf
    Untracked:  output/Deliverables/Goal2/DifferentialAbundance/lefse/boxplots/paired_Clostridium_hiranonis_vs_Timepoint_Placebo_Goal2.pdf
    Untracked:  output/Deliverables/Goal2/DifferentialAbundance/lefse/boxplots/paired_Clostridium_sp_CAG_58_vs_Timepoint_Probiotic_Goal2.pdf
    Untracked:  output/Deliverables/Goal2/DifferentialAbundance/lefse/boxplots/paired_Collinsella_intestinalis_vs_Timepoint_Placebo_Goal2.pdf
    Untracked:  output/Deliverables/Goal2/DifferentialAbundance/lefse/boxplots/paired_Dorea_sp_CAG_317_vs_Timepoint_Probiotic_Goal2.pdf
    Untracked:  output/Deliverables/Goal2/DifferentialAbundance/lefse/lefser_goal2_Probiotic_plot.pdf
    Untracked:  output/Deliverables/Goal2/DifferentialAbundance/maaslin/
    Untracked:  output/Deliverables/Goal3/
    Untracked:  output/Deliverables/NMDS_StressPlot_BL.pdf

Unstaged changes:
    Modified:   analysis/01_index.Rmd
    Modified:   analysis/02_Exploratory.Rmd
    Modified:   analysis/03_Goal1.Rmd
    Deleted:    analysis/04_Conclusiones.Rmd
    Modified:   analysis/04_Goal2.Rmd
    Modified:   analysis/05_Goal3.Rmd
    Modified:   analysis/_site.yml
    Deleted:    analysis/about.Rmd
    Modified:   output/Deliverables/Exploratory/AlphaDiversity/RarefactionCurve_timepoint.pdf
    Modified:   output/Deliverables/Exploratory/BetaDiversity/Adonis.xls
    Modified:   output/Deliverables/Exploratory/BetaDiversity/NMDS_StressPlot.pdf
    Modified:   output/Deliverables/Exploratory/BetaDiversity/Ordination_AllSamples.pdf
    Modified:   output/Deliverables/Exploratory/ExploratoryHeatmap_allSamples.pdf
    Modified:   output/Deliverables/Exploratory/FinalMetadata.xlsx
    Modified:   output/Deliverables/Exploratory/PhylumAbundance_RA.xls
    Modified:   output/Deliverables/Exploratory/PhylumAbundance_counts.xls
    Modified:   output/Deliverables/Exploratory/Spaghetti_Efaecium.pdf
    Modified:   output/Deliverables/Exploratory/Spaghetti_Efaecium_BLexcluded.pdf
    Modified:   output/Deliverables/Exploratory/SpeciesAbundance_RA.xls
    Modified:   output/Deliverables/Exploratory/SpeciesAbundance_counts.xls
    Modified:   output/Deliverables/Exploratory/StackedBarplots_Species.pdf
    Modified:   output/Deliverables/Exploratory/ps_metaphlan3.biom
    Modified:   output/Deliverables/Goal1/AlphaDiversity/AlphaDiversity_Timepoint_goal1_only.pdf
    Modified:   output/Deliverables/Goal1/BetaDiversity/NMDS_Ordination_vs_Timepoint.pdf
    Modified:   output/Deliverables/Goal1/DifferentialAbundance/lefse/boxplots/Allisonella_histaminiformans_vs_Timepoint_Goal1.pdf
    Modified:   output/Deliverables/Goal1/DifferentialAbundance/lefse/boxplots/Bifidobacterium_longum_vs_Timepoint_Goal1.pdf
    Modified:   output/Deliverables/Goal1/DifferentialAbundance/lefse/boxplots/Bifidobacterium_pseudocatenulatum_vs_Timepoint_Goal1.pdf
    Modified:   output/Deliverables/Goal1/DifferentialAbundance/lefse/boxplots/Bifidobacterium_pullorum_vs_Timepoint_Goal1.pdf
    Modified:   output/Deliverables/Goal1/DifferentialAbundance/lefse/boxplots/Collinsella_tanakaei_vs_Timepoint_Goal1.pdf
    Modified:   output/Deliverables/Goal1/DifferentialAbundance/lefse/boxplots/paired_Allisonella_histaminiformans_vs_Timepoint_Goal1.pdf
    Modified:   output/Deliverables/Goal1/DifferentialAbundance/lefse/boxplots/paired_Bifidobacterium_longum_vs_Timepoint_Goal1.pdf
    Modified:   output/Deliverables/Goal1/DifferentialAbundance/lefse/boxplots/paired_Bifidobacterium_pseudocatenulatum_vs_Timepoint_Goal1.pdf
    Modified:   output/Deliverables/Goal1/DifferentialAbundance/lefse/boxplots/paired_Bifidobacterium_pullorum_vs_Timepoint_Goal1.pdf
    Modified:   output/Deliverables/Goal1/DifferentialAbundance/lefse/boxplots/paired_Collinsella_tanakaei_vs_Timepoint_Goal1.pdf
    Modified:   output/Deliverables/Goal1/DifferentialAbundance/lefse/lefser_goal1_plot.pdf
    Deleted:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/function/all_results.tsv
    Deleted:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/function/fitted.rds
    Deleted:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/function/genes/all_results.tsv
    Deleted:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/function/genes/fitted.rds
    Deleted:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/function/genes/maaslin2.log
    Deleted:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/function/genes/ranef.rds
    Deleted:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/function/genes/residuals.rds
    Deleted:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/function/genes/significant_results.tsv
    Deleted:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/function/maaslin2.log
    Deleted:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/function/pathways/all_results.tsv
    Deleted:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/function/pathways/fitted.rds
    Deleted:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/function/pathways/maaslin2.log
    Deleted:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/function/pathways/ranef.rds
    Deleted:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/function/pathways/residuals.rds
    Deleted:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/function/pathways/significant_results.tsv
    Deleted:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/function/ranef.rds
    Deleted:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/function/residuals.rds
    Deleted:    output/Deliverables/Goal1/DifferentialAbundance/maaslin/function/significant_results.tsv
    Modified:   output/Deliverables/Goal1/DifferentialAbundance/maaslin/genus/maaslin2.log
    Modified:   output/Deliverables/Goal1/DifferentialAbundance/maaslin/species/maaslin2.log
    Modified:   output/Deliverables/Goal2/AlphaDiversity/AlphaDiversity_Timepoint_goal2_Fortiflora_samples.pdf
    Modified:   output/Deliverables/Goal2/AlphaDiversity/AlphaDiversity_Timepoint_goal2_Placebo_samples.pdf
    Modified:   output/Deliverables/Goal2/AlphaDiversity/AlphaDiversity_Timepoint_goal2_all_samples.pdf
    Modified:   output/Deliverables/Goal2/BetaDiversity/BrayDistance_by_Treatment_goal2.pdf
    Modified:   output/Deliverables/Goal2/BetaDiversity/Goal2_Adonis.xls
    Modified:   output/Deliverables/Goal2/BetaDiversity/NMDS_Goal2_StressPlot.pdf
    Modified:   output/Deliverables/Goal2/BetaDiversity/NMDS_Ordination_vs_Timepoint_and_treatment.pdf
    Deleted:    output/Deliverables/Goal2/DifferentialAbundance/lefse/boxplots/Blautia_sp_CAG_257_vs_Timepoint_Fortiflora_Goal2.pdf
    Modified:   output/Deliverables/Goal2/DifferentialAbundance/lefse/boxplots/Clostridium_hiranonis_vs_Timepoint_Placebo_Goal2.pdf
    Deleted:    output/Deliverables/Goal2/DifferentialAbundance/lefse/boxplots/Clostridium_sp_CAG_58_vs_Timepoint_Fortiflora_Goal2.pdf
    Modified:   output/Deliverables/Goal2/DifferentialAbundance/lefse/boxplots/Collinsella_intestinalis_vs_Timepoint_Placebo_Goal2.pdf
    Deleted:    output/Deliverables/Goal2/DifferentialAbundance/lefse/boxplots/Dorea_sp_CAG_317_vs_Timepoint_Fortiflora_Goal2.pdf
    Deleted:    output/Deliverables/Goal2/DifferentialAbundance/lefse/boxplots/paired_Blautia_sp_CAG_257_vs_Timepoint_Fortiflora_Goal2.pdf
    Deleted:    output/Deliverables/Goal2/DifferentialAbundance/lefse/boxplots/paired_Clostridium_hiranonis_vs_Timepoint_Placebo_Goal1.pdf
    Deleted:    output/Deliverables/Goal2/DifferentialAbundance/lefse/boxplots/paired_Clostridium_sp_CAG_58_vs_Timepoint_Fortiflora_Goal2.pdf
    Deleted:    output/Deliverables/Goal2/DifferentialAbundance/lefse/boxplots/paired_Collinsella_intestinalis_vs_Timepoint_Placebo_Goal1.pdf
    Deleted:    output/Deliverables/Goal2/DifferentialAbundance/lefse/boxplots/paired_Dorea_sp_CAG_317_vs_Timepoint_Fortiflora_Goal2.pdf
    Deleted:    output/Deliverables/Goal2/DifferentialAbundance/lefse/lefser_goal2_Fortiflora_plot.pdf
    Modified:   output/Deliverables/Goal2/DifferentialAbundance/lefse/lefser_goal2_Placebo_plot.pdf
    Modified:   output/Deliverables/NMDS_Goal1_StressPlot.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/01_index.Rmd</code>) and HTML (<code>docs/01_index.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TreeTopUnder/Purina_N1H_2022/blob/74b0908090ff8e0bc963c89a8444a189c23e53da/analysis/01_index.Rmd" target="_blank">74b0908</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-10-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TreeTopUnder/Purina_N1H_2022/74b0908090ff8e0bc963c89a8444a189c23e53da/docs/01_index.html" target="_blank">74b0908</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-10-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TreeTopUnder/Purina_N1H_2022/blob/59ac17e3c27e6b80b71def7e067af5282980f584/analysis/01_index.Rmd" target="_blank">59ac17e</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-09-13
</td>
<td>
initial commits
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TreeTopUnder/Purina_N1H_2022/59ac17e3c27e6b80b71def7e067af5282980f584/docs/01_index.html" target="_blank">59ac17e</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-09-13
</td>
<td>
initial commits
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="about-the-project" class="section level1">
<h1>About the project</h1>
<ul>
<li>Goal: Gut Microbiome characterization in leishmaniosis-affected dogs under treatment. Longitudinal effect of probiotics after treatment initiation.</li>
<li>Context: Canine leishmaniosis (CL)</li>
<li>Specimen Type: Faecal (stool, rectal palpation)</li>
<li>Sequencing: Whole DNA shotgun metagenomics. Illumina paired-end data fastq (125/150x2, NovaSeq/HiSeq).</li>
<li>Sequencing provider: MacroGEN (Korea)</li>
<li>Sample: Multi-centric longitudinal study with CL-affected&amp; treated dogs that are supplemented with probiotic. Four timepoints available (M0, M1, M4 and M6) for each dog.</li>
<li>Clinical design: Multicentric, prospective, controlled, randomized and blinded. Twenty-four dogs with leishmaniosis. All dogs will assigned to one of the two groups: treatment with probiotic (Group 1) or treatment with placebo (Group 2) at the end of the first month of leishmaniosis treatment. The treatment for both groups will be Glucantime (50-100 mg/kg/every 12h or 24h during 1 month) plus allopurinol (10 mg/kg/12h), during 6 months, that could be reduced to every 24h if xanthinuria is detected in any of the follow up rechecks.</li>
<li>Sample Size: The study aimed to recruit n=24 dogs with a follow-up with 6 months (4 timepoints each). Ten (n=10) dogs have been successfully recruited with complete follow-up. Eight (n=8) dogs provided only baseline samples and were not treated. Effective sample size for longitudinal analysis is n=10. Clinical trial goals:
<ol style="list-style-type: decimal">
<li>Evaluate changes in the intestinal microbiome composition of dogs with leishmaniosis at diagnosis and during the treatment (Glucantime and allopurinol)</li>
<li>Evaluate the effects of probiotic supplementation on the gut microbiome, after Glucantime treatment interruption, and on the clinical progression of dogs treated for leishmaniosis.</li>
<li>Assess the gut microbiome of leishmaniasis-affected dogs, before treatment.</li>
</ol></li>
</ul>
<p>Response variable(s):</p>
<ul>
<li>Primary Variables: Timepoint(Goal 1 &amp; 2), Probiotic(Goal 2).</li>
<li><pre><code>Secondary Variables: gender, animal_name, lifestyle, Leish_stage</code></pre></li>
</ul>
<p>Software: * Trimmomatic/Kraken2 * Bowtie2/IGC * Biobakery3::Metaphlan/HUMANN * R::phyloseq/R::vegan/R::lefser/R::maaslin/R::r_statix</p>
</div>
<div id="methods" class="section level1">
<h1>Methods</h1>
<div id="gut-metagenomic-sequencing" class="section level2">
<h2>Gut Metagenomic Sequencing</h2>
<p>(Details on sample collection, shipment, DNA extraction) Extracted DNA was submitted to Macrogen(Macrogen, South Korea) for sequencing, with a median depth of 10 million reads 150x2 in paired-end configuration.</p>
</div>
<div id="sequence-data-analysis" class="section level2">
<h2>Sequence Data Analysis</h2>
<p>Raw data is obtained in fastq format. Data have been acquired through a whole DNA shotgun metagenomic sequencing using an Illumina HiSeq 150x2 bp instrument and sequencing kit. Using these data, several analysis have been performed:</p>
<ul>
<li>Quality Control using trimmomatic. Low quality and adapter sequence have either been removed or trimmed from the dataset.</li>
<li>Gene Marker-based taxonomical quantification using Metaphlan3. (v201901b)</li>
<li>Gene Catalog Mapping using Integrated Gene Catalog (IGC)</li>
<li>Functional quantification using humann3.(v201901b)</li>
<li>When possible, Analysis results have been packaged into phyloseq object for further analysis.</li>
</ul>
<p>In this way, we obtain three data packets. Each of them contains a different aspecte that can be obtained from the sequencing data, including taxonomical quantification down to the species level, gene richness/diversity, and gene function. This will allow us to calculate bot alpha- and beta-diversity estimators.</p>
</div>
</div>
<div id="results" class="section level1">
<h1>Results</h1>
<div id="metadata" class="section level2">
<h2>Metadata</h2>
<p><em>Deliverables: FinalMetadata.xls y WMGSDataAvailability.xls</em></p>
<p>Each of the FASTQ files is labeled with an identifier like <em>1234_{0,1,4,6}</em>. There is WMGS data available for 46 samples, including a positive control (Mock Community/Zymo). Of these, 8 samples correspond to animals who only contributed baseline samples. The rest of samples correspond to animals that contributed 4 samples in total at baseline, M1, M4 and M6. Individual animals are identified by an identifier variable (<em>Host_name</em>). Relevant sample variables are</p>
<ul>
<li>Host_name: Animal identifier.</li>
<li>Timepoint: Timepoint of the sample (BL/M1/M4/M9)</li>
<li>Sex: Sex of the animal</li>
<li>neutered: Boolean relative to animal neuterin</li>
<li>Lifestyle: Animal’s lifestyle (Countryside/Urban/Mixed)</li>
<li>LeishVet_stage: Leishmaniasis disease clinical stage at the moment of sampling (0, I, II, III, IV).</li>
</ul>
<p>In this dataset we are going to discard all non prokaryotic taxa. (Kingdom = Archaea or Kingdom = Bacteria). No significant differences in throughput vs Timepoint are found.</p>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<pre><code>            Timepoint
Host_name    BL M1 M4 M6
  BALOO       1  1  1  1
  BANDIDO     1  1  1  1
  CHARLIE     1  1  1  1
  GELATINA    1  1  1  1
  IMPERATORE  1  1  1  1
  LEO         1  1  1  1
  LOLO        0  0  0  0
  MAX         1  0  0  0
  NANI        1  1  1  1
  OLAF        1  0  0  0
  PERLA       1  0  0  0
  RUDDY       1  1  1  1
  TARA        1  0  0  0
  THOR        1  0  0  0
  TINY LEE    1  1  1  1
  TOTO        1  1  1  1</code></pre>
<p>Dataset characteristics are:</p>
<ul>
<li>45 gut microbiome samples: Of these, there are 15 baseline samples, and 10 samples at M1, M4, M6. 10 Animals provide complete follow-up samples.</li>
<li>1 <em>mock</em> sample with known composition</li>
<li>A total of 194 taxas.</li>
<li>Taxonomic annotation down to the species level</li>
</ul>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 197 taxa and 45 samples ]
sample_data() Sample Data:       [ 45 samples by 47 sample variables ]
tax_table()   Taxonomy Table:    [ 197 taxa by 7 taxonomic ranks ]</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] metar_0.1.5      testthat_3.1.4   forcats_0.5.1    stringr_1.4.1   
 [5] dplyr_1.0.10     purrr_0.3.5      readr_2.1.2      tidyr_1.2.1     
 [9] tibble_3.1.8     ggplot2_3.3.6    tidyverse_1.3.2  phylosmith_1.0.6
[13] here_1.0.1       pander_0.6.5     phyloseq_1.38.0  workflowr_1.7.0 

loaded via a namespace (and not attached):
  [1] utf8_1.2.2             lme4_1.1-30            tidyselect_1.2.0      
  [4] htmlwidgets_1.5.4      grid_4.1.1             Rtsne_0.16            
  [7] devtools_2.4.4         munsell_0.5.0          codetools_0.2-18      
 [10] units_0.8-0            future_1.27.0          miniUI_0.1.1.1        
 [13] withr_2.5.0            colorspace_2.0-3       Biobase_2.54.0        
 [16] logger_0.2.2           knitr_1.39             rstudioapi_0.13       
 [19] stats4_4.1.1           ggsignif_0.6.3         rJava_1.0-6           
 [22] listenv_0.8.0          git2r_0.30.1           GenomeInfoDbData_1.2.7
 [25] polyclip_1.10-0        bit64_4.0.5            farver_2.1.1          
 [28] rhdf5_2.38.1           rprojroot_2.0.3        parallelly_1.32.1     
 [31] coda_0.19-4            vctrs_0.4.2            generics_0.1.3        
 [34] xfun_0.31              R6_2.5.1               GenomeInfoDb_1.30.1   
 [37] arm_1.12-2             graphlayouts_0.8.0     bitops_1.0-7          
 [40] rhdf5filters_1.6.0     cachem_1.0.6           assertthat_0.2.1      
 [43] networkD3_0.4          vroom_1.5.7            promises_1.2.0.1      
 [46] scales_1.2.1           ggraph_2.0.5           googlesheets4_1.0.0   
 [49] merTools_0.5.2         gtable_0.3.1           globals_0.15.1        
 [52] processx_3.7.0         xlsx_0.6.5             tidygraph_1.2.1       
 [55] rlang_1.0.6            splines_4.1.1          rstatix_0.7.0         
 [58] gargle_1.2.0           broom_1.0.0            yaml_2.3.5            
 [61] reshape2_1.4.4         abind_1.4-5            modelr_0.1.8          
 [64] backports_1.4.1        httpuv_1.6.5           tools_4.1.1           
 [67] usethis_2.1.6          logging_0.10-108       ellipsis_0.3.2        
 [70] selbal_0.1.0           jquerylib_0.1.4        biomformat_1.22.0     
 [73] RColorBrewer_1.1-3     proxy_0.4-27           BiocGenerics_0.40.0   
 [76] sessioninfo_1.2.2      Rcpp_1.0.9             plyr_1.8.7            
 [79] zlibbioc_1.40.0        classInt_0.4-7         RCurl_1.98-1.8        
 [82] ps_1.7.1               prettyunits_1.1.1      ggpubr_0.4.0          
 [85] viridis_0.6.2          urlchecker_1.0.1       S4Vectors_0.32.4      
 [88] haven_2.5.0            ggrepel_0.9.1          cluster_2.1.3         
 [91] fs_1.5.2               furrr_0.3.0            DataExplorer_0.8.2    
 [94] magrittr_2.0.3         data.table_1.14.2      reprex_2.0.1          
 [97] mvtnorm_1.1-3          googledrive_2.0.0      whisker_0.4           
[100] pkgload_1.3.0          xlsxjars_0.6.1         hms_1.1.2             
[103] patchwork_1.1.1        mime_0.12              evaluate_0.15         
[106] xtable_1.8-4           broom.mixed_0.2.9.4    readxl_1.4.0          
[109] IRanges_2.28.0         gridExtra_2.3          compiler_4.1.1        
[112] KernSmooth_2.23-20     crayon_1.5.1           minqa_1.2.4           
[115] htmltools_0.5.3        mgcv_1.8-40            later_1.3.0           
[118] tzdb_0.3.0             RcppParallel_5.1.5     lubridate_1.8.0       
[121] DBI_1.1.3              tweenr_1.0.2           dbplyr_2.2.1          
[124] MASS_7.3-58            boot_1.3-28            sf_1.0-7              
[127] wesanderson_0.3.6      Matrix_1.4-1           ade4_1.7-19           
[130] car_3.1-0              brio_1.1.3             permute_0.9-7         
[133] cli_3.4.1              parallel_4.1.1         igraph_1.3.5          
[136] pkgconfig_2.0.3        getPass_0.2-2          xml2_1.3.3            
[139] foreach_1.5.2          bslib_0.4.0            blme_1.0-5            
[142] multtest_2.50.0        XVector_0.34.0         rvest_1.0.2           
[145] callr_3.7.1            digest_0.6.29          vegan_2.6-4           
[148] Biostrings_2.62.0      rmarkdown_2.14         cellranger_1.1.0      
[151] shiny_1.7.2            nloptr_2.0.3           lifecycle_1.0.3       
[154] nlme_3.1-158           jsonlite_1.8.0         Rhdf5lib_1.16.0       
[157] carData_3.0-5          desc_1.4.1             viridisLite_0.4.1     
[160] fansi_1.0.3            pillar_1.8.1           lattice_0.20-45       
[163] fastmap_1.1.0          httr_1.4.4             pkgbuild_1.3.1        
[166] survival_3.3-1         glue_1.6.2             remotes_2.4.2         
[169] iterators_1.0.14       bit_4.0.4              ggforce_0.3.3         
[172] class_7.3-20           stringi_1.7.8          sass_0.4.2            
[175] profvis_0.3.7          memoise_2.0.1          e1071_1.7-11          
[178] ape_5.6-2             </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
