<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marc Noguera-Julian, PhD. TreeTopUnder" />

<meta name="date" content="2022-10-02" />

<title>Effect of Leishmaniasis treatment</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Purina_N1H_2022</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="01_index.html">Home</a>
</li>
<li>
  <a href="02_Exploratory.html">Exploratory</a>
</li>
<li>
  <a href="03_Goal1.html">Goal 1</a>
</li>
<li>
  <a href="04_Goal2.html">Goal 2</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Effect of Leishmaniasis treatment</h1>
<h4 class="author">Marc Noguera-Julian, PhD. TreeTopUnder</h4>
<h4 class="date">2022-10-02</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-10-02
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Purina_N1H_2022/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220906code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220906)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220906code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220906)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTreeTopUnderPurinaN1H2022tree59ac17e3c27e6b80b71def7e067af5282980f584targetblank59ac17ea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/TreeTopUnder/Purina_N1H_2022/tree/59ac17e3c27e6b80b71def7e067af5282980f584" target="_blank">59ac17e</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomTreeTopUnderPurinaN1H2022tree59ac17e3c27e6b80b71def7e067af5282980f584targetblank59ac17ea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/TreeTopUnder/Purina_N1H_2022/tree/59ac17e3c27e6b80b71def7e067af5282980f584" target="_blank">59ac17e</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/001_metadata_check.Rmd
    Ignored:    code/003_metadata_vars_selection.Rmd
    Ignored:    code/006_DataPreprocessing.Rmd
    Ignored:    code/006_DataPreprocessing.html
    Ignored:    code/008_PopulateMRE.Rmd
    Ignored:    code/output/
    Ignored:    data/.DS_Store
    Ignored:    data/mreObject_feces.rds
    Ignored:    data/mreObject_goal1.rds
    Ignored:    data/mreObject_goal2.rds
    Ignored:    data/mreObject_goal3.rds
    Ignored:    data/processed/
    Ignored:    data/ps_metaphlan_feces.rds
    Ignored:    data/raw/
    Ignored:    output/.DS_Store
    Ignored:    output/Deliverables/.DS_Store
    Ignored:    output/Deliverables/Goal1/.DS_Store
    Ignored:    output/Deliverables/Goal3/
    Ignored:    output/mre_data/.DS_Store
    Ignored:    output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/
    Ignored:    output/mre_data/feces/

Untracked files:
    Untracked:  analysis/02_Exploratory.Rmd
    Untracked:  analysis/03_Goal1.Rmd
    Untracked:  analysis/04_Conclusiones.Rmd
    Untracked:  analysis/04_Goal2.Rmd
    Untracked:  analysis/05_Goal3.Rmd
    Untracked:  output/Deliverables/Exploratory/
    Untracked:  output/Deliverables/Goal1/AlphaDiversity/
    Untracked:  output/Deliverables/Goal1/DifferentialAbundance/
    Untracked:  output/Deliverables/Goal2/

Unstaged changes:
    Modified:   analysis/01_index.Rmd
    Deleted:    analysis/02_Exploratory_v1.Rmd
    Deleted:    analysis/03_Goal1_v1.Rmd
    Deleted:    analysis/04_Conclusiones_v1.Rmd
    Modified:   analysis/_site.yml
    Deleted:    output/Deliverables/Adonis.xls
    Deleted:    output/Deliverables/AlphaDiversity_Timepoint_goal1_only.pdf
    Deleted:    output/Deliverables/AlphaDiversity_treatment.pdf
    Deleted:    output/Deliverables/AlphaDiversity_treatment_dry_only.pdf
    Deleted:    output/Deliverables/AlphaDiversity_treatment_w1_only.pdf
    Deleted:    output/Deliverables/AlphaDiversity_treatment_w2_only.pdf
    Deleted:    output/Deliverables/BrayCurtis_Distances.xls
    Deleted:    output/Deliverables/ExploratoryHeatmap_allSamples.pdf
    Deleted:    output/Deliverables/FinalMetadata.xlsx
    Deleted:    output/Deliverables/GeneRichness_Estimation.xls
    Modified:   output/Deliverables/Goal1/BetaDiversity/Goal1_Adonis.xls
    Modified:   output/Deliverables/Goal1/BetaDiversity/NMDS_Ordination_vs_Timepoint.pdf
    Deleted:    output/Deliverables/Goal1/BetaDiversity/sGoal1_Adonis.xls
    Deleted:    output/Deliverables/Goal1/DifferentialAbundanceAllisonella_vs_Timepoint_Goal1.pdf
    Deleted:    output/Deliverables/Goal1/DifferentialAbundanceBifidobacterium_vs_Timepoint_Goal1.pdf
    Deleted:    output/Deliverables/Goal1/DifferentialAbundanceCollinsella_vs_Timepoint_Goal1.pdf
    Deleted:    output/Deliverables/Goal1/Goal1_Adonis.xls
    Modified:   output/Deliverables/NMDS_Goal1_StressPlot.pdf
    Deleted:    output/Deliverables/NMDS_StressPlot.pdf
    Deleted:    output/Deliverables/Ordination_AllSamples.pdf
    Deleted:    output/Deliverables/PhylumAbundance_RA.xls
    Deleted:    output/Deliverables/PhylumAbundance_counts.xls
    Deleted:    output/Deliverables/RarefactionCurve_timepoint.pdf
    Deleted:    output/Deliverables/SpeciesAbundance_RA.xls
    Deleted:    output/Deliverables/SpeciesAbundance_counts.xls
    Deleted:    output/Deliverables/StackedBarplots_Species.pdf
    Deleted:    output/Deliverables/ps_metaphlan3.biom

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<p>The goal of this subanalysis is to characterize changes in the gut microbiome of Leishmaniasis-affected dogs after 1 month of treatment with Allopurinol and Glucantime(R). For that, data for 10 different animals is available at BL, prior to treatment, and at M1. All animals underwent Leishamaniasis treatment and there were no control animals nor samples. Thus, we’ll work with a subset of 20 samples and the primary variable we will use is <em>Timepoint</em>.</p>
<p>Note: sample sizes for each of the timepoints (n=10) are low and have limited statistical power due to under-recruitment at the clinical stage. Results may be affected.</p>
<div id="alpha-diversity" class="section level2">
<h2>Alpha-Diversity</h2>
<p>Alpha Diversity in each of the samples has been previously calculated. The questions are:</p>
<ul>
<li>Does Alpha-diversity longitudinally change after 1 month of Leishmaniasis treatment?</li>
<li>Does Gene Richness longitudinally change after 1 month of Leishmaniasis treatment?</li>
</ul>
<p><em>Deliverables: AlphaDiversity.pdf</em></p>
<p>Results indicate that there are no difference between M1 and BL samples using either Chao1, Shannon or GeneRichness alpha-diversity estimators. Generally speaking, Chao1 and Shannon and similar estimators based on marker-gene taxonomical classification have poor representativity. Gene richness, on the other hand, evaluate the number of different genes which is more suitable for whole metagenome data on a gene-detection base.</p>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<p><img src="figure/03_Goal1.Rmd/alpha-diversity%20vs%20treatment-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="beta-diversity" class="section level2">
<h2>Beta-Diversity</h2>
<div id="ordination-analysis" class="section level3">
<h3>Ordination Analysis</h3>
<p>We will use the same ordination method than in previous exploratory section but, in this case, only applied to BL and M0 sample to detect significant changes related to the effect of Allopurinol + Glucantime treatment, discarding the effect of other variables that may act as confounders.</p>
<p><em>Deliverables: Goal1/BetaDiversity/NMDS_Ordination_vs_Timepoint.pdf</em></p>
<p>No effect of (<em>Timepoint</em>) is observed or significant in the global composition of the gut microbiome of treated animals. Animal identity, however, explains 61% of the microbiome composition. This is expected in longitudinal data.</p>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<pre><code>Run 0 stress 0.1767754 
Run 1 stress 0.149536 
... New best solution
... Procrustes: rmse 0.1166604  max resid 0.4114427 
Run 2 stress 0.1812884 
Run 3 stress 0.1493793 
... New best solution
... Procrustes: rmse 0.038812  max resid 0.1051225 
Run 4 stress 0.149536 
... Procrustes: rmse 0.03885512  max resid 0.1069193 
Run 5 stress 0.1494701 
... Procrustes: rmse 0.03591243  max resid 0.1033198 
Run 6 stress 0.203298 
Run 7 stress 0.245139 
Run 8 stress 0.1493793 
... New best solution
... Procrustes: rmse 2.014624e-06  max resid 6.55875e-06 
... Similar to previous best
Run 9 stress 0.1493793 
... Procrustes: rmse 2.160725e-06  max resid 6.990051e-06 
... Similar to previous best
Run 10 stress 0.1972797 
Run 11 stress 0.149536 
... Procrustes: rmse 0.03880553  max resid 0.1067651 
Run 12 stress 0.1825951 
Run 13 stress 0.1805401 
Run 14 stress 0.1493793 
... Procrustes: rmse 1.154854e-06  max resid 3.035847e-06 
... Similar to previous best
Run 15 stress 0.1494401 
... Procrustes: rmse 0.03484375  max resid 0.1018176 
Run 16 stress 0.1493793 
... Procrustes: rmse 2.476569e-06  max resid 8.20261e-06 
... Similar to previous best
Run 17 stress 0.1493793 
... Procrustes: rmse 1.956634e-06  max resid 6.845098e-06 
... Similar to previous best
Run 18 stress 0.14944 
... Procrustes: rmse 0.03485484  max resid 0.1018559 
Run 19 stress 0.1495361 
... Procrustes: rmse 0.03885576  max resid 0.1069217 
Run 20 stress 0.1494401 
... Procrustes: rmse 0.03479756  max resid 0.1016872 
*** Solution reached</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre><code>[1] TRUE</code></pre>
<p><img src="figure/03_Goal1.Rmd/Global%20Ordination%20Beta%20Diversity%20-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="differential-abundance" class="section level2">
<h2>Differential Abundance</h2>
<p>We have seen that the is no structural change in the composition related to leishmaniasis treatment. We are interested, now, if there is any change in specific bacterial species that, even if minor and not causing structural changes, may be relevant in relationship with the effect of the treatment. For that purpose we’ll use different methods:</p>
<ul>
<li>LEFsER: This is the R implementation of LEFsE, a biomarker discovery software designed for microbiome data.</li>
<li>MaasLin:</li>
<li>Wilcoxon: Standard Wilcoxon test used on CLR-transformed data.</li>
</ul>
<p>Note that within this group we have n=10 for each group being compared (10 at samples BL and 10 samples at M1). Therefore, the dataset and design is under-powered to detect small differences.</p>
<div id="lefse-vs.-treatment" class="section level3">
<h3>LEFsE vs. Treatment</h3>
<p>LEFsE results show an increase in M1, of three different <em>Bifidobacterium</em> species(). One limitation of LEFsE is that, internally, non-parametric tests are non-paired, so it does not take into account the paired design of the study. When the paired design is applied, significance(p&lt;0.05) of these species vs Timepoint is lost (see deliverable boxplots), although trends are mantained. Notably, <em>Bifodbacterium</em> species are not foundin any BL samples, but are found in 4 M1 samples. In addition, changes in these species are located in 4 different animals and not the other 6, which also undergo the same treatment, thus pointing at the possiblity that these changes are related to an unobserved factor other than treatment.</p>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<p><img src="figure/03_Goal1.Rmd/Timepoint%20DA%20chunk1%20Species-1.png" width="672" style="display: block; margin: auto;" /></p>
<table>
<colgroup>
<col width="7%" />
<col width="23%" />
<col width="16%" />
<col width="14%" />
<col width="17%" />
<col width="20%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Host_name</th>
<th align="right">Bifidobacterium_pseudocatenulatum</th>
<th align="right">Bifidobacterium_longum</th>
<th align="right">Collinsella_tanakaei</th>
<th align="right">Bifidobacterium_pullorum</th>
<th align="right">Allisonella_histaminiformans</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">RUDDY</td>
<td align="right">1852784</td>
<td align="right">733367</td>
<td align="right">9463</td>
<td align="right">586</td>
<td align="right">2674</td>
</tr>
<tr class="even">
<td align="left">NANI</td>
<td align="right">397276</td>
<td align="right">758521</td>
<td align="right">156090</td>
<td align="right">11850</td>
<td align="right">100</td>
</tr>
<tr class="odd">
<td align="left">LEO</td>
<td align="right">15514</td>
<td align="right">25984</td>
<td align="right">173229</td>
<td align="right">7169</td>
<td align="right">617</td>
</tr>
<tr class="even">
<td align="left">BALOO</td>
<td align="right">2766</td>
<td align="right">27815</td>
<td align="right">9612</td>
<td align="right">1170</td>
<td align="right">622</td>
</tr>
<tr class="odd">
<td align="left">TINY LEE</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">TOTO</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">GELATINA</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">BANDIDO</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">CHARLIE</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">IMPERATORE</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
</div>
<div id="maaslin-vs-treatment" class="section level3">
<h3>MaasLin vs Treatment</h3>
<p>MaasLin method allows for the accounting of other variables in the test design.</p>
<p>Results: Using MaasLin2 there are no significant hits at species nor genus level</p>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<pre><code>[1] &quot;Warning: Deleting existing log file: output/Deliverables/Goal1/DifferentialAbundance/maaslin/species/maaslin2.log&quot;
2022-10-02 10:39:06 INFO::Writing function arguments to log file
2022-10-02 10:39:06 INFO::Verifying options selected are valid
2022-10-02 10:39:06 INFO::Determining format of input files
2022-10-02 10:39:06 INFO::Input format is data samples as rows and metadata samples as rows
2022-10-02 10:39:06 INFO::Formula for random effects: expr ~ (1 | Host_name)
2022-10-02 10:39:06 INFO::Formula for fixed effects: expr ~  Timepoint
2022-10-02 10:39:06 INFO::Filter data based on min abundance and min prevalence
2022-10-02 10:39:06 INFO::Total samples in data: 20
2022-10-02 10:39:06 INFO::Min samples required with min abundance for a feature not to be filtered: 2.000000
2022-10-02 10:39:06 INFO::Total filtered features: 134
2022-10-02 10:39:06 INFO::Filtered feature names from abundance and prevalence filtering:
2022-10-02 10:39:06 INFO::Total filtered features with variance filtering: 0
2022-10-02 10:39:06 INFO::Filtered feature names from variance filtering:
2022-10-02 10:39:06 INFO::Running selected normalization method: TSS
2022-10-02 10:39:06 INFO::Applying z-score to standardize continuous metadata
2022-10-02 10:39:06 INFO::Running selected transform method: LOG
2022-10-02 10:39:06 INFO::Running selected analysis method: LM
2022-10-02 10:39:06 INFO::Creating cluster of 4 R processes
2022-10-02 10:39:12 INFO::Counting total values for each feature
2022-10-02 10:39:12 WARNING::Deleting existing residuals file: output/Deliverables/Goal1/DifferentialAbundance/maaslin/species/residuals.rds
2022-10-02 10:39:12 INFO::Writing residuals to file output/Deliverables/Goal1/DifferentialAbundance/maaslin/species/residuals.rds
2022-10-02 10:39:12 WARNING::Deleting existing fitted file: output/Deliverables/Goal1/DifferentialAbundance/maaslin/species/fitted.rds
2022-10-02 10:39:12 INFO::Writing fitted values to file output/Deliverables/Goal1/DifferentialAbundance/maaslin/species/fitted.rds
2022-10-02 10:39:12 WARNING::Deleting existing ranef file: output/Deliverables/Goal1/DifferentialAbundance/maaslin/species/ranef.rds
2022-10-02 10:39:12 INFO::Writing extracted random effects to file output/Deliverables/Goal1/DifferentialAbundance/maaslin/species/ranef.rds
2022-10-02 10:39:12 INFO::Writing all results to file (ordered by increasing q-values): output/Deliverables/Goal1/DifferentialAbundance/maaslin/species/all_results.tsv
2022-10-02 10:39:12 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.250000 ) to file (ordered by increasing q-values): output/Deliverables/Goal1/DifferentialAbundance/maaslin/species/significant_results.tsv
2022-10-02 10:39:12 INFO::Writing heatmap of significant results to file: output/Deliverables/Goal1/DifferentialAbundance/maaslin/species/heatmap.pdf
[1] &quot;There are no associations to plot!&quot;
2022-10-02 10:39:12 INFO::Writing association plots (one for each significant association) to output folder: output/Deliverables/Goal1/DifferentialAbundance/maaslin/species
[1] &quot;There are no associations to plot!&quot;</code></pre>
<pre><code> [1] &quot;9255_1&quot; &quot;5235_1&quot; &quot;9648_0&quot; &quot;3510_1&quot; &quot;7909_1&quot; &quot;2282_1&quot; &quot;5235_0&quot; &quot;3790_1&quot;
 [9] &quot;9356_0&quot; &quot;7344_1&quot; &quot;9045_1&quot; &quot;9532_0&quot; &quot;7344_0&quot; &quot;7909_0&quot; &quot;9353_0&quot; &quot;2282_0&quot;
[17] &quot;I0&quot;     &quot;I1&quot;     &quot;C0&quot;     &quot;C1&quot;    </code></pre>
<pre><code>[1] &quot;Warning: Deleting existing log file: output/Deliverables/Goal1/DifferentialAbundance/maaslin/genus/maaslin2.log&quot;
2022-10-02 10:39:14 INFO::Writing function arguments to log file
2022-10-02 10:39:14 INFO::Verifying options selected are valid
2022-10-02 10:39:14 INFO::Determining format of input files
2022-10-02 10:39:14 INFO::Input format is data samples as rows and metadata samples as rows
2022-10-02 10:39:14 INFO::Formula for random effects: expr ~ (1 | Host_name)
2022-10-02 10:39:14 INFO::Formula for fixed effects: expr ~  Timepoint
2022-10-02 10:39:14 INFO::Filter data based on min abundance and min prevalence
2022-10-02 10:39:14 INFO::Total samples in data: 20
2022-10-02 10:39:14 INFO::Min samples required with min abundance for a feature not to be filtered: 2.000000
2022-10-02 10:39:14 INFO::Total filtered features: 48
2022-10-02 10:39:14 INFO::Filtered feature names from abundance and prevalence filtering:
2022-10-02 10:39:14 INFO::Total filtered features with variance filtering: 0
2022-10-02 10:39:14 INFO::Filtered feature names from variance filtering:
2022-10-02 10:39:14 INFO::Running selected normalization method: TSS
2022-10-02 10:39:14 INFO::Applying z-score to standardize continuous metadata
2022-10-02 10:39:14 INFO::Running selected transform method: LOG
2022-10-02 10:39:14 INFO::Running selected analysis method: LM
2022-10-02 10:39:14 INFO::Creating cluster of 4 R processes
2022-10-02 10:39:19 INFO::Counting total values for each feature
2022-10-02 10:39:19 WARNING::Deleting existing residuals file: output/Deliverables/Goal1/DifferentialAbundance/maaslin/genus/residuals.rds
2022-10-02 10:39:19 INFO::Writing residuals to file output/Deliverables/Goal1/DifferentialAbundance/maaslin/genus/residuals.rds
2022-10-02 10:39:19 WARNING::Deleting existing fitted file: output/Deliverables/Goal1/DifferentialAbundance/maaslin/genus/fitted.rds
2022-10-02 10:39:19 INFO::Writing fitted values to file output/Deliverables/Goal1/DifferentialAbundance/maaslin/genus/fitted.rds
2022-10-02 10:39:19 WARNING::Deleting existing ranef file: output/Deliverables/Goal1/DifferentialAbundance/maaslin/genus/ranef.rds
2022-10-02 10:39:19 INFO::Writing extracted random effects to file output/Deliverables/Goal1/DifferentialAbundance/maaslin/genus/ranef.rds
2022-10-02 10:39:19 INFO::Writing all results to file (ordered by increasing q-values): output/Deliverables/Goal1/DifferentialAbundance/maaslin/genus/all_results.tsv
2022-10-02 10:39:19 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.250000 ) to file (ordered by increasing q-values): output/Deliverables/Goal1/DifferentialAbundance/maaslin/genus/significant_results.tsv
2022-10-02 10:39:19 INFO::Writing heatmap of significant results to file: output/Deliverables/Goal1/DifferentialAbundance/maaslin/genus/heatmap.pdf
[1] &quot;There are no associations to plot!&quot;
2022-10-02 10:39:19 INFO::Writing association plots (one for each significant association) to output folder: output/Deliverables/Goal1/DifferentialAbundance/maaslin/genus
[1] &quot;There are no associations to plot!&quot;</code></pre>
</div>
</div>
<div id="functional-analysis" class="section level2">
<h2>Functional Analysis</h2>
<p>One of the advantages of using metagenomic sequencing is the possibility to profile gene abundance in addition to taxonomical composition. this is done using HUMANN3 software. Briefly, this software performs the analysis in three stages:</p>
<ul>
<li>Detection of the bacterial species that are present in the sample above threshold.</li>
<li>Download of full genomes of detected species and indexes of the sample-specific genomes and gene annotation. In this manner, we obtain a table of relative abundances for UniRef Gene families that are annotated using KEGG and eggNOG ontologies.</li>
<li>Read mapping to sample-specific full-genome. Quantification of gene relative abundance and annotation to pathways if possible using MetaCyc.</li>
</ul>
<p>From these analysis, we’ll use MetaCYC pathway relative abundance quantification to compare sample groups. We run MaasLin algorithm both at the MetaCyc pathway and UniRef levels using standard TSS normalization and log transformation. No significant hits were found for Timepoint in this dataset.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] metar_0.1.5                 Maaslin2_1.8.0             
 [3] lefser_1.4.0                SummarizedExperiment_1.24.0
 [5] Biobase_2.54.0              GenomicRanges_1.46.1       
 [7] GenomeInfoDb_1.30.1         IRanges_2.28.0             
 [9] S4Vectors_0.32.4            BiocGenerics_0.40.0        
[11] MatrixGenerics_1.6.0        matrixStats_0.62.0         
[13] ggforce_0.3.3               ggplot2_3.3.6              
[15] testthat_3.1.4              dplyr_1.0.9                
[17] patchwork_1.1.1             phyloseq_1.38.0            
[19] workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] coda_0.19-4            ragg_1.2.2             tidyr_1.2.0           
  [4] bit64_4.0.5            knitr_1.39             multcomp_1.4-19       
  [7] DelayedArray_0.20.0    wesanderson_0.3.6      data.table_1.14.2     
 [10] DataExplorer_0.8.2     RCurl_1.98-1.8         generics_0.1.3        
 [13] callr_3.7.1            TH.data_1.1-1          usethis_2.1.6         
 [16] future_1.27.0          correlation_0.8.1      bit_4.0.4             
 [19] tzdb_0.3.0             httpuv_1.6.5           assertthat_0.2.1      
 [22] xfun_0.31              jquerylib_0.1.4        rJava_1.0-6           
 [25] evaluate_0.15          promises_1.2.0.1       DEoptimR_1.0-11       
 [28] fansi_1.0.3            igraph_1.3.5           DBI_1.1.3             
 [31] htmlwidgets_1.5.4      hash_2.2.6.2           paletteer_1.4.0       
 [34] purrr_0.3.4            ellipsis_0.3.2         ggpubr_0.4.0          
 [37] backports_1.4.1        insight_0.18.0         permute_0.9-7         
 [40] prismatic_1.1.0        libcoin_1.0-9          vctrs_0.4.1           
 [43] remotes_2.4.2          here_1.0.1             abind_1.4-5           
 [46] cachem_1.0.6           withr_2.5.0            robustbase_0.95-0     
 [49] vroom_1.5.7            vegan_2.6-2            prettyunits_1.1.1     
 [52] getopt_1.20.3          cluster_2.1.3          ape_5.6-2             
 [55] crayon_1.5.1           pkgconfig_2.0.3        labeling_0.4.2        
 [58] tweenr_1.0.2           nlme_3.1-158           statsExpressions_1.3.2
 [61] pkgload_1.3.0          blme_1.0-5             devtools_2.4.4        
 [64] rlang_1.0.6            globals_0.15.1         lifecycle_1.0.2       
 [67] miniUI_0.1.1.1         sandwich_3.0-2         selbal_0.1.0          
 [70] rprojroot_2.0.3        polyclip_1.10-0        datawizard_0.4.1      
 [73] Matrix_1.4-1           lpsymphony_1.22.0      carData_3.0-5         
 [76] Rhdf5lib_1.16.0        boot_1.3-28            zoo_1.8-10            
 [79] whisker_0.4            processx_3.7.0         parameters_0.18.1     
 [82] bitops_1.0-7           getPass_0.2-2          rhdf5filters_1.6.0    
 [85] Biostrings_2.62.0      afex_1.1-1             stringr_1.4.1         
 [88] coin_1.4-2             arm_1.12-2             parallelly_1.32.1     
 [91] rstatix_0.7.0          ggsignif_0.6.3         scales_1.2.1          
 [94] memoise_2.0.1          magrittr_2.0.3         plyr_1.8.7            
 [97] zlibbioc_1.40.0        compiler_4.1.1         RColorBrewer_1.1-3    
[100] lme4_1.1-30            cli_3.4.1              ade4_1.7-19           
[103] XVector_0.34.0         urlchecker_1.0.1       lmerTest_3.1-3        
[106] listenv_0.8.0          pbapply_1.5-0          ps_1.7.1              
[109] MASS_7.3-58            mgcv_1.8-40            tidyselect_1.1.2      
[112] stringi_1.7.8          forcats_0.5.1          textshaping_0.3.6     
[115] highr_0.9              yaml_2.3.5             grid_4.1.1            
[118] sass_0.4.2             tools_4.1.1            parallel_4.1.1        
[121] rstudioapi_0.13        foreach_1.5.2          logging_0.10-108      
[124] git2r_0.30.1           optparse_1.7.3         gridExtra_2.3         
[127] farver_2.1.1           digest_0.6.29          shiny_1.7.2           
[130] networkD3_0.4          Rcpp_1.0.9             car_3.1-0             
[133] broom_1.0.0            xlsx_0.6.5             performance_0.9.1     
[136] later_1.3.0            httr_1.4.4             effectsize_0.7.0      
[139] colorspace_2.0-3       brio_1.1.3             fs_1.5.2              
[142] splines_4.1.1          rematch2_2.1.2         xlsxjars_0.6.1        
[145] multtest_2.50.0        sessioninfo_1.2.2      systemfonts_1.0.4     
[148] xtable_1.8-4           jsonlite_1.8.0         nloptr_2.0.3          
[151] zeallot_0.1.0          modeltools_0.2-23      R6_2.5.1              
[154] profvis_0.3.7          broom.mixed_0.2.9.4    pillar_1.8.1          
[157] htmltools_0.5.3        mime_0.12              glue_1.6.2            
[160] fastmap_1.1.0          minqa_1.2.4            codetools_0.2-18      
[163] pkgbuild_1.3.1         pcaPP_2.0-2            mvtnorm_1.1-3         
[166] furrr_0.3.0            utf8_1.2.2             lattice_0.20-45       
[169] bslib_0.4.0            tibble_3.1.8           logger_0.2.2          
[172] numDeriv_2016.8-1.1    biglm_0.9-2.1          survival_3.3-1        
[175] rmarkdown_2.14         desc_1.4.1             biomformat_1.22.0     
[178] merTools_0.5.2         munsell_0.5.0          rhdf5_2.38.1          
[181] GenomeInfoDbData_1.2.7 iterators_1.0.14       ggstatsplot_0.9.3     
[184] reshape2_1.4.4         gtable_0.3.1           bayestestR_0.12.1     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
