---
site: workflowr::wflow_site
title: "Metagenomic Analysis of Canine gut microbiome in Leishmaniasis"
author: "Marc Noguera-Julian, PhD. TreeTopUnder"
date: "`r format(Sys.time(), '%Y-%m-%d')`"
header-includes:
- \usepackage{fancyhdr}
- \usepackage{lipsum}
- \pagestyle{fancy}
- \fancyhead[CO,CE]{2022-PurinaLeish-WMGS}
- \fancyfoot[LO,RE]{Marc Noguera-Julian, TreeTopUnder}
- \fancyhead[LE,RO]{\thepage}
- \fancyfoot[CO,CE]{}
- \usepackage[default]{sourcesanspro}
- \usepackage[T1]{fontenc}
- \usepackage{eso-pic,graphicx,transparent}
mainfont: Helvetica
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = here::here())
```

# About the project

  * Goal:  Gut Microbiome characterization in leishmaniosis-affected dogs under treatment. Longitudinal effect of probiotics after treatment initiation.
  * Context: Canine leishmaniosis (CL)
  * Specimen Type: Faecal (stool, rectal palpation)
  * Sequencing: Whole DNA shotgun metagenomics. Illumina paired-end data fastq (125/150x2, NovaSeq/HiSeq). 
  * Sequencing provider: MacroGEN (Korea)
  * Sample: Multi-centric longitudinal study with CL-affected& treated dogs that are supplemented with probiotic. Four timepoints available (M0, M1, M4 and M6) for each dog. 
  * Clinical design: Multicentric, prospective, controlled, randomized and blinded.  Twenty-four dogs with leishmaniosis. All dogs will assigned to one of the two groups: treatment with probiotic (Group 1) or treatment with placebo (Group 2) at the end of the first month of leishmaniosis treatment. The treatment for both groups will be Glucantime (50-100 mg/kg/every 12h or 24h during 1 month) plus allopurinol (10 mg/kg/12h), during 6 months, that could be reduced to every 24h if xanthinuria is detected in any of the follow up rechecks.
  * Sample Size: The study aimed to recruit n=24 dogs with a follow-up with 6 months (4 timepoints each). Ten (n=10) dogs have been successfully recruited with complete follow-up. Eight (n=8) dogs provided only baseline samples and were not treated. Effective sample size for longitudinal analysis is n=10.
Clinical trial goals: 
    1.	Evaluate changes in the intestinal microbiome composition of dogs with leishmaniosis at diagnosis and during the treatment (Glucantime and allopurinol)
    2.	Evaluate the effects of probiotic supplementation on the gut microbiome, after Glucantime treatment interruption, and on the clinical progression of dogs treated for leishmaniosis.
    3.	Assess the gut microbiome of leishmaniasis-affected dogs, before treatment.

Response variable(s): 

  * Primary Variables: Timepoint(Goal 1 & 2), Probiotic(Goal 2).
  *     Secondary Variables: gender, animal_name, lifestyle, Leish_stage

Software: 
  * Trimmomatic/Kraken2
  * Bowtie2/IGC
  * Biobakery3::Metaphlan/HUMANN
  * R::phyloseq/R::vegan/R::lefser/R::maaslin/R::r_statix

# Methods

## Sequence Data Analysis

Raw data is obtained in fastq format. Data have been acquired through a whole DNA shotgun metagenomic sequencing using an Illumina HiSeq 150x2 bp instrument and sequencing kit. Using these data, several analysis have been performed:

- Quality Control using trimmomatic. Low quality and adapter sequence have either been removed or trimmed from the dataset.
- Gene Marker-based taxonomical quantification using Metaphlan3. (v201901b)
- Gene Catalog Mapping using Integrated Gene Catalog (IGC)
- Functional quantification using humann3.(v201901b)
- Each analysis result has been packaged into a phyloseq object for further analysis.

In this way, we obtain three data packets. Each of them contains a different aspecte that can be obtained from the sequencing data, including taxonomical quantification down to the species level, gene richness/diversity, and gene function. This will allow us to calculate bot alpha- and beta-diversity estimators.

# Results

## Metadata

*Deliverables: FinalMetadata.xls y WMGSDataAvailability.xls*

Each of the FASTQ files is labeled with an identifier like *1234_{0,1,4,6}*. There is WMGS data available for 46 samples, including a positive control (Mock Community/Zymo). Of these, 8 samples correspond to animals who only contributed baseline samples. The rest of samples correspond to animals that contributed 4 samples in total at baseline, M1, M4 and M6. Individual animals are identified by an identifier variable (*Host_name*). Relevant sample variables are

-   Host_name: Animal identifier.
-   Timepoint: Timepoint of the sample (BL/M1/M4/M9)
-   Sex: Sex of the animal
-   neutered: Boolean relative to animal neuterin
-   Lifestyle: Animal's lifestyle (Countryside/Urban/Mixed)
-   LeishVet_stage: Leishmaniasis disease clinical stage at the moment of sampling (0, I, II, III, IV).


In this dataset we are going to discard all non prokaryotic taxa. (Kingdom = Archaea or Kingdom = Bacteria). No significant differences in throughput vs Timepoint are found.

```{r Consolidacion de datos y metadatos, echo=F, message=FALSE, warning=FALSE, paged.print=FALSE}
suppressWarnings(require(phyloseq))
suppressWarnings(require(pander))
suppressWarnings(require(here))
suppressWarnings(require(phylosmith))
suppressWarnings(require(tidyverse))
suppressWarnings(require(ggplot2))
### Load data and create MRE Object
devtools::load_all("~/Documents/Work/Development/WMGSPipeline/")### Change to require/library/load local package

myMREObject<-readRDS("data/processed/mreObject.rds")
metadata_df<-metar::get_meta(myMREObject)
rownames(metadata_df)<-metadata_df$SampleID
mock_samples<-c("MOCKCOMMUNITY")
myMREObject<-metadata_df %>% 
  dplyr::filter(! SampleID %in% mock_samples) %>% 
  dplyr::select(SampleID) %>% c() %>% .$SampleID %>% 
  metar::filter_samples(mre = myMREObject,sample_ids = .)

myMREObject@wd<-"output/mre_data/feces/"
saveRDS(myMREObject,file="data/mreObject_feces.rds")



# control_samples<-c("11802AAC-AFF-BA","11803AAC-D6306-1","11804AAC-Cneg","11805AAC-D6306-2","11788AAC-AFF-B-Ext")
# low_throughput_samples<-c("S01993AAD-M105")
# metadata_df[,"sample_type"]<-"specimen"
# metadata_df[metadata_df$SampleID %in% control_samples,"sample_type"]<-"Control"
metadata_df[metadata_df$SampleID %in% mock_samples,"sample_type"]<-"mock"
# metadata_df <-merge(metadata_df,track,by="row.names")
# rownames(metadata_df)<-metadata_df$SampleID
# 
#  # xtabs(~neutered+treatment,data=metadata_df) %>% chisq.test()
# sample_data(ps_metaphlan)<-metadata_df
my_datatable<-readRDS("data/raw/CatalogMapping/IGC/dataTable.rds")

metadata_df<-my_datatable %>% 
  dplyr::select(SampleID,InitialReads) %>% 
  dplyr::distinct() %>% 
  dplyr::left_join(metadata_df) %>%
  dplyr::rename(QFPassReads=InitialReads) 

# metadata_df%>% 
#   ggstatsplot::ggbetweenstats(x=Timepoint,y=QFPassReads,title = "Throughput vs Timepoint",
#                               type="nonparametric",bf.message = F,
#                               outlier.tagging = T,outlier.label = SampleID)


xlsx::write.xlsx(metadata_df,"output/Deliverables/Exploratory/FinalMetadata.xlsx")
metadata_df<-metadata_df[! is.na(metadata_df$SampleID),]
rownames(metadata_df)<-metadata_df$SampleID
ps_metaphlan<-metar::get_phyloseq(myMREObject,type="metaphlan")
ps_metaphlan<-phyloseq::prune_samples(! sample_names(ps_metaphlan) %in% mock_samples, ps_metaphlan)
ps_metaphlan<-phyloseq::subset_taxa(ps_metaphlan,Kingdom %in% c("Bacteria","Archaea"))
saveRDS(ps_metaphlan,file="data/ps_metaphlan_feces.rds")

table(metadata_df %>% dplyr::select(Host_name,Timepoint)) 

### Save objects for longitudinal analysis
myMREObject<-readRDS("data/processed/mreObject.rds")
aux_DF<-table(metadata_df %>% 
               dplyr::select(Host_name,Timepoint)) %>% 
  as.data.frame() %>% 
  dplyr::group_by(Host_name) %>% 
  dplyr::summarise(counts=sum(Freq)) %>% 
  dplyr::filter(counts==4)

myMREObject <- metadata_df %>% 
  dplyr::filter(Host_name %in% aux_DF$Host_name) %>%
  dplyr::filter(Timepoint %in% c("BL","M1")) %>% .[,"SampleID"] %>% 
  metar::filter_samples(mre = myMREObject,sample_ids = .)
myMREObject@wd<-"output/mre_data/goal1/"
saveRDS(myMREObject,file="data/mreObject_goal1.rds")

### Dataset for goal2
myMREObject<-readRDS("data/processed/mreObject.rds")
aux_DF<-table(metadata_df %>% 
               dplyr::select(Host_name,Timepoint)) %>% 
  as.data.frame() %>% 
  dplyr::group_by(Host_name) %>% 
  dplyr::summarise(counts=sum(Freq)) %>% 
  dplyr::filter(counts==4)

myMREObject <- metadata_df %>% 
  dplyr::filter(Host_name %in% aux_DF$Host_name) %>%
  dplyr::filter(Timepoint %in% c("M1","M4")) %>% .[,"SampleID"] %>% 
  metar::filter_samples(mre = myMREObject,sample_ids = .)
myMREObject@wd<-"output/mre_data/goal2/"
saveRDS(myMREObject,file="data/mreObject_goal2.rds")

### Dataset for goal3
myMREObject<-readRDS("data/processed/mreObject.rds")
aux_DF<-table(metadata_df %>% 
               dplyr::select(Host_name,Timepoint)) %>% 
  as.data.frame() %>% 
  dplyr::group_by(Host_name) %>% 
  dplyr::summarise(counts=sum(Freq)) %>% 
  dplyr::filter(counts==4)

myMREObject <- metadata_df %>% 
  dplyr::filter(Host_name %in% aux_DF$Host_name) %>%
  dplyr::filter(Timepoint %in% c("BL")) %>% .[,"SampleID"] %>% 
  metar::filter_samples(mre = myMREObject,sample_ids = .)
myMREObject@wd<-"output/mre_data/goal3/"
saveRDS(myMREObject,file="data/mreObject_goal3.rds")
```

Dataset characteristics are:

-   45 gut microbiome samples:  Of these, there are 15 baseline samples, and 10 samples at M1, M4, M6. 10 Animals provide complete follow-up samples.
-   1  *mock* sample with known composition
-   A total of `r phyloseq::ntaxa(ps_metaphlan)` taxas.
-   Taxonomic annotation down to the species level


```{r data summary,eval=T, echo=F, message=FALSE, warning=FALSE, paged.print=FALSE}
ps_metaphlan<-readRDS("data/ps_metaphlan_feces.rds")
ps_metaphlan
# data.frame(sample_data(ps_metaphlan)) %>% DataExplorer::create_report(output_file = "docs/MetadataExplorer.html")
```
