---
site: workflowr::wflow_site
title: "Metagenomic Analysis of Canine gut microbiome in Leishmaniasis"
author: "Marc Noguera-Julian, PhD. TreeTopUnder"
date: "`r format(Sys.time(), '%Y-%m-%d')`"
header-includes:
- \usepackage{fancyhdr}
- \usepackage{lipsum}
- \pagestyle{fancy}
- \fancyhead[CO,CE]{2022-PurinaLeish-WMGS}
- \fancyfoot[LO,RE]{Marc Noguera-Julian, TreeTopUnder}
- \fancyhead[LE,RO]{\thepage}
- \fancyfoot[CO,CE]{}
- \usepackage[default]{sourcesanspro}
- \usepackage[T1]{fontenc}
- \usepackage{eso-pic,graphicx,transparent}
mainfont: Helvetica
output: html_document
---

```{r setup, include=TRUE,echo=F, message=FALSE, warning=FALSE, paged.print=FALSE, tidy=F}
knitr::opts_chunk$set(echo = FALSE)
knitr::opts_knit$set(root.dir = here::here())
```

The goal of this subanalysis is to characterize changes in the gut microbiome of Leishmaniasis-affected dogs after 1 month of treatment with Allopurinol and Glucantime(R). For that, data for 10 different animals is available at BL, prior to treatment, and at M1. All animals underwent Leishamaniasis treatment and there were no control animals nor samples. Thus, we'll work with a subset of 20 samples and the primary variable we will use is *Timepoint*. 

Note: sample sizes for each of the timepoints (n=10) are low and have limited statistical power due to under-recruitment at the clinical stage. Results may be affected.

## Alpha-Diversity

Alpha Diversity in each of the samples has been previously calculated. The questions are:

- Does Alpha-diversity longitudinally change after 1 month of Leishmaniasis treatment?
- Does Gene Richness longitudinally change after 1 month of Leishmaniasis treatment?

*Deliverables: AlphaDiversity.pdf*

Results indicate that there are no difference between M1 and BL samples using either Chao1, Shannon or GeneRichness alpha-diversity estimators.

```{r alfa-diversity vs treatment, include=TRUE,echo=F, message=FALSE, warning=FALSE, paged.print=FALSE, tidy=F,fig.height = 15}
suppressMessages(require(phyloseq))
suppressMessages(require(patchwork))
suppressMessages(require(dplyr))
### Load data and create MRE Object
devtools::load_all("~/Documents/Work/Development/WMGSPipeline/")### Change to require/library/load local package

myMREObject<-readRDS("data/mreObject_goal1.rds")
my_datatable<-readRDS("data/raw/CatalogMapping/IGC/dataTable.rds")
mymin<-min(my_datatable$NumberMappedReads) #
my_datatable<-my_datatable %>% 
  dplyr::filter(ReadCountReal==6000000) %>% 
  dplyr::filter(SampleID %in% get_meta(myMREObject)$SampleID) %>% 
  dplyr::select(SampleID,GeneNumber)

x.2.0 <- myMREObject@taxa@metaphlan@phyloseq_sec
x.2.0.species<-x.2.0
# x.2.0.species = transform_sample_counts(x.2.0.species, function(x) ((x/sum(x))))

# get_meta(myMREObject) %>% 
#   dplyr::select(Timepoint,Host_name) %>% table()

plot_data<-phyloseq::estimate_richness(x.2.0.species) %>% 
  tibble::rownames_to_column() %>% 
  dplyr::mutate(rowname=gsub("X","",rowname)) %>% 
  dplyr::left_join(get_meta(myMREObject),by=c("rowname"="SampleID")) %>% 
  dplyr::rename(SampleID=rowname) %>% 
  dplyr::select(SampleID, Timepoint, Host_name, Chao1,Shannon) %>% 
  dplyr::left_join(my_datatable,by="SampleID") %>% 
  reshape2::melt() 

p0<-
  plot_data %>% 
  dplyr::arrange(Host_name) %>% 
  dplyr::group_by(Host_name,variable) %>% 
  ggstatsplot::grouped_ggwithinstats(x=Timepoint,y=value,grouping.var = variable,type="non-parametric",centrality.plotting=F,
                                      annotation.args = list(title = "treatment: M1 vs BL",
                                                         theme = ggplot2::theme(
                                                                plot.subtitle = ggplot2::element_text(size = 13),
                                                                plot.title = ggplot2::element_text(size = 14)
                                                                )
                                                         ),
                                  plotgrid.args = list(nrow = 3))
p0
# pT<- p0 / p1 / p2
ggplot2::ggsave(p0,filename="output/Deliverables/AlphaDiversity_Timepoint_goal1_only.pdf",height=15,width=7)
# ggplot2::ggsave(p1,filename="output/Deliverables/AlphaDiversity_treatment_w1_only.pdf",height=15,width=7)
# ggplot2::ggsave(p2,filename="output/Deliverables/AlphaDiversity_treatment_w2_only.pdf",height=15,width=7)
# ggplot2::ggsave(pT,filename="output/Deliverables/AlphaDiversity_treatment.pdf",height=15,width=7)
```

## Beta-Diversity

### Ordination Analysis

We will use the same ordination method than in previous exploratory section but, in this case, only applied to BL and M0 sample to detect significant changes related to the effect of Allopurinol + Glucantime treatment.

*Deliverables: BetaDiversityEvolution_per_group.pdf*

No effect of  (*Timepoint*) is observed or significant in the global composition of the gut mcirobiome of treated animals. Animal identify, however, explains 61% of the microbiome composition.

```{r Paired Beta Diversity per variable, include=TRUE,echo=F, message=FALSE, warning=FALSE, paged.print=FALSE, tidy=F,fig.height = 15}
suppressWarnings(require(phyloseq))
suppressWarnings(require(dplyr))
suppressWarnings(require(ggplot2))
suppressWarnings(require(ggforce))
suppressWarnings(require(patchwork))
set.seed(1234)

devtools::load_all("~/Documents/Work/Development/WMGSPipeline/")
myMREObject<-readRDS("data/mreObject_goal1.rds")
x.2.0<-myMREObject@taxa@metaphlan@phyloseq_sec
set.seed(1234)
x.2.0.species<-x.2.0
x.2.0.species<-transform_sample_counts(x.2.0.species, function(x) ((x/sum(x))))
sampleData<-data.frame(sample_data(x.2.0.species))
rownames(sampleData)<-sampleData$SampleID
sample_data(x.2.0.species)<-sampleData
(myOrd<-phyloseq::ordinate(x.2.0.species,method="NMDS",distance="bray",trymax=100))

pdf("output/Deliverables/NMDS_Goal1_StressPlot.pdf")
vegan::stressplot(myOrd)
dev.off()

#### PERMUTATIONAL ANOVA (adonis()) on Bray-Curtus distance
x.6.0.explanatory<-sampleData[,c("Timepoint","SampleID","Host_name")]
x.6.0.response<-phyloseq::distance((x.2.0.species),method="bray")
myAdonis<-vegan::adonis2(x.6.0.response~Timepoint,data=x.6.0.explanatory)
file.remove("output/Deliverables/Goal1/BetaDiversity/Goal1_Adonis.xls")
xlsx::write.xlsx(file = "output/Deliverables/Goal1/BetaDiversity/Goal1_Adonis.xls",sheetName = "Timepoint",vegan::adonis2(x.6.0.response~Timepoint,data=x.6.0.explanatory))
### Perform ANOVA for non-NA in treatment
xlsx::write.xlsx(file = "output/Deliverables/Goal1/BetaDiversity/sGoal1_Adonis.xls",sheetName = "Host_name",vegan::adonis2(x.6.0.response~Host_name,data=x.6.0.explanatory),append=T)

p.4.0.samples=plot_ordination(x.2.0.species,myOrd)

### By treatment
p0<-p.4.0.samples  + geom_point(aes(color=Timepoint)) +ggtitle("Unconstrained NMDS(Bray) by Timepoint | Goal1")+
  theme(panel.border=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.line=element_line(colour="black"),axis.text.x=element_text(size=16),axis.text.y=element_text(size=16))+
  theme(axis.title.x=element_text(size=14),axis.title.y=element_text(size=14))+
  theme(legend.text=element_text(size=12,face="italic"),legend.title=element_text(size=12))+
  theme(plot.title=element_text(lineheight=1,face="bold",size=15))+stat_ellipse(alpha=1,mapping=aes(color=Timepoint))+
  theme_classic() + theme(axis.line.x = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.line.y = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.text.x = element_text(size = 10), 
                                    axis.text.y = element_text(size = 10),
                                    axis.title.x = element_text(size = 10, 
                                        face = "bold"), axis.title.y = element_text(size = 10, 
                                        face = "bold"), legend.title = element_text(size = 10, 
                                        face = "bold"), legend.text = element_text(size = 8), 
                                        legend.spacing.x = unit(0.005, "npc"), 
                                        legend.key.size = unit(4,"mm"), legend.background = element_rect(fill = (alpha = 0))) + 
        labs(x = "NMDS Dimension 1", y = "NMDS Dimension 2") + scale_fill_brewer(palette="Set1")+scale_color_brewer(palette="Set1")+
        guides(fill = guide_legend(override.aes = list(size = 4)))

ggsave(plot=p0,filename="output/Deliverables/Goal1/BetaDiversity/NMDS_Ordination_vs_Timepoint.pdf",height=12,width=7)
```


## Differential Abundance


### vs. Treatment
En este análisis se responderá a las preguntas relacionadas con el cambio concretos, a nivel de género bacteriano, de la composición del microbioma en relación a la intervención nutricional


```{r Timepoint DA chunk1 Genus, include=TRUE,echo=F, message=FALSE, warning=FALSE, paged.print=FALSE, tidy=F,eval=F}
suppressMessages(require(dplyr))
suppressMessages(require(phyloseq))
suppressMessages(require(lefser))
suppressMessages(require(ggplot2))
suppressMessages(require(patchwork))
ps_silva<-readRDS("data/ps_silva.rds")
set.seed(1234)
### Species Level

devtools::load_all("~/Documents/Work/Development/WMGSPipeline/")
myMREObject<-readRDS("data/mreObject_goal1.rds")
x.2.0<-myMREObject@taxa@metaphlan@phyloseq_sec
set.seed(1234)
x.2.0.species<-x.2.0
# x.2.0.species<-transform_sample_counts(x.2.0.species, function(x) ((x/sum(x))))


### Genus Level
x.2.0 <-x.2.0.species %>% 
  # phyloseq::filter_taxa(., function (x) sum(x > 0) > 2, TRUE) %>%
  phyloseq::rarefy_even_depth(.) %>% 
  phyloseq::tax_glom(taxrank="Genus",NArm = F)
  # microbiome::transform(transform="compositional")

se_ps<-SummarizedExperiment::SummarizedExperiment(assays = list(counts=as.matrix(otu_table(x.2.0))),
                                                  colData=sample_data(x.2.0),
                                                  rowData=tax_table(x.2.0))
aux_res<-lefser(expr=se_ps,groupCol="Timepoint",blockCol = NULL,lda.threshold = 1)
aux_res<-merge(aux_res,tax_table(x.2.0),by.x="Names",by.y="row.names")%>% 
  dplyr::select(Names,Phylum,Class,Order,Family,Genus,scores) %>% 
  dplyr::mutate(Class=ifelse(is.na(Class),Phylum,Class)) %>% 
  dplyr::mutate(Order=ifelse(is.na(Order),Class,Order)) %>% 
  dplyr::mutate(Family=ifelse(is.na(Family),Order,Family)) %>% 
  dplyr::mutate(Genus=ifelse(is.na(Genus),paste0(Family,"_spp"),Genus)) %>% 
  dplyr::filter(! is.na(Genus)) %>% 
  dplyr::mutate(otu=Names) %>% 
  dplyr::mutate(Names=Genus) %>% 
  dplyr::select(otu,Names,scores) %>% 
  dplyr::mutate(Names = forcats::fct_reorder(Names, scores)) %>% 
  dplyr::mutate(Timepoint=ifelse(scores>0,"BL","M1"))
p0<-aux_res %>% 
    ggplot(.,aes(x=Names,y=scores,fill=Timepoint))+geom_bar(stat="identity")+coord_flip()+
    theme_classic()+ggtitle("Month1 vs BL")+
    scale_fill_brewer(palette="Dark2")

for(otu_name in aux_res$otu){
  my_ps <- x.2.0 
  title <- aux_res %>% 
    dplyr::filter(otu==otu_name) 
  my_ps %>% 
  phyloseq::psmelt()%>% 
  dplyr::filter(OTU==otu_name) %>% 
  ggstatsplot::ggwithinstats(data = .,x=Timepoint,y=Abundance,
                                       xlab=NULL,ylab=NULL,centrality.plotting=F,
                                       bf.message=F,title=title$Names,
                                       method="non-parametric",palette="Dark2") %>% 
   ggsave(.,filename=paste0("output/Deliverables/Goal1/DifferentialAbundance",title$Names,"_vs_Timepoint_Goal1.pdf"),width=7, height=7)
}
```
```{r Timepoint DA chunk1 Genus, include=TRUE,echo=F, message=FALSE, warning=FALSE, paged.print=FALSE, tidy=F,eval=F}
suppressMessages(require(dplyr))
suppressMessages(require(phyloseq))
suppressMessages(require(lefser))
suppressMessages(require(ggplot2))
suppressMessages(require(patchwork))
ps_silva<-readRDS("data/ps_silva.rds")
set.seed(1234)
### Species Level

devtools::load_all("~/Documents/Work/Development/WMGSPipeline/")
myMREObject<-readRDS("data/mreObject_goal1.rds")
x.2.0<-myMREObject@taxa@metaphlan@phyloseq_sec
set.seed(1234)
x.2.0.species<-x.2.0
# x.2.0.species<-transform_sample_counts(x.2.0.species, function(x) ((x/sum(x))))


### Genus Level
x.2.0 <-x.2.0.species %>% 
  # phyloseq::filter_taxa(., function (x) sum(x > 0) > 2, TRUE) %>%
  phyloseq::rarefy_even_depth(.) %>% 
  phyloseq::tax_glom(taxrank="Genus",NArm = F)
  # microbiome::transform(transform="compositional")

se_ps<-SummarizedExperiment::SummarizedExperiment(assays = list(counts=as.matrix(otu_table(x.2.0))),
                                                  colData=sample_data(x.2.0),
                                                  rowData=tax_table(x.2.0))
aux_res<-lefser(expr=se_ps,groupCol="Timepoint",blockCol = NULL,lda.threshold = 1)
aux_res<-merge(aux_res,tax_table(x.2.0),by.x="Names",by.y="row.names")%>% 
  dplyr::select(Names,Phylum,Class,Order,Family,Genus,scores) %>% 
  dplyr::mutate(Class=ifelse(is.na(Class),Phylum,Class)) %>% 
  dplyr::mutate(Order=ifelse(is.na(Order),Class,Order)) %>% 
  dplyr::mutate(Family=ifelse(is.na(Family),Order,Family)) %>% 
  dplyr::mutate(Genus=ifelse(is.na(Genus),paste0(Family,"_spp"),Genus)) %>% 
  dplyr::filter(! is.na(Genus)) %>% 
  dplyr::mutate(otu=Names) %>% 
  dplyr::mutate(Names=Genus) %>% 
  dplyr::select(otu,Names,scores) %>% 
  dplyr::mutate(Names = forcats::fct_reorder(Names, scores)) %>% 
  dplyr::mutate(Timepoint=ifelse(scores>0,"BL","M1"))
p0<-aux_res %>% 
    ggplot(.,aes(x=Names,y=scores,fill=Timepoint))+geom_bar(stat="identity")+coord_flip()+
    theme_classic()+ggtitle("Month1 vs BL")+
    scale_fill_brewer(palette="Dark2")

for(otu_name in aux_res$otu){
  my_ps <- x.2.0 
  title <- aux_res %>% 
    dplyr::filter(otu==otu_name) 
  my_ps %>% 
  phyloseq::psmelt()%>% 
  dplyr::filter(OTU==otu_name) %>% 
  ggstatsplot::ggwithinstats(data = .,x=Timepoint,y=Abundance,
                                       xlab=NULL,ylab=NULL,centrality.plotting=F,
                                       bf.message=F,title=title$Names,
                                       method="non-parametric",palette="Dark2") %>% 
   ggsave(.,filename=paste0("output/Deliverables/Goal1/DifferentialAbundance",title$Names,"_vs_Timepoint_Goal1.pdf"),width=7, height=7)
}
```


### Variable *treatment*

Comparativa no paramétrica + LDA entre la abundancia de diferentes géneros bacterianos entre puntos temporales t1 vs t0, estratificada por grupos de tratamiento (d/w & d/w1/w2) y la valoración del *effect size* en la diferencia mediante el paquete *lefser*.

*Deliverables: lefser_t1_vs_t0_d\_samples.pdf, lefser_t1_vs_t0_w1_samples.pdf, lefser_t1_vs_t0_w2_samples.pdf y boxplots independientes para cada género positivo al test*

Se observan taxas diferencialmente abundantes entre puntos temporales, especialmente en los grupos de tratamiento w2 y w **lefser**(<http://www.bioconductor.org/packages/release/bioc/html/lefser.html>):

-   Mayor abundancia en perros después de haber seguido dieta húmeda(w2): *Blautia, Dorea, Escherichia Shigella, grupo R.torques i R.gnavus*
-   Menor abundancia en perros después de haber seguido dieta húmeda(w2): *Peptoclostridium, Megamonas, Prevotella, Alloprevotella, Sutterella*

```{r DA vs time_point all treatment groups, include=TRUE,echo=F, message=FALSE, warning=FALSE, paged.print=FALSE, tidy=F}
suppressMessages(require(dplyr))
suppressMessages(require(phyloseq))
suppressMessages(require(lefser))
suppressMessages(require(ggplot2))
suppressMessages(require(patchwork))
ps_silva<-readRDS("data/ps_silva.rds")
set.seed(1234)
### Genus Level
### We are really interested in comparing t1 vs t0, separately for each d,w1,w2 and d/w

x.2.0 <-ps_silva %>% 
  phyloseq::subset_samples(dog_name!="Charlotte") %>% 
  phyloseq::tax_glom(.,taxrank = "Genus",NArm = F) %>%
  # phyloseq::filter_taxa(., function (x) sum(x > 0) > 2, TRUE) %>%
  phyloseq::rarefy_even_depth(.)


### d samples
x.2.0 <- x.2.0 %>% 
  phyloseq::subset_samples(treatment=="d")
  # microbiome::transform(transform="compositional")

se_ps<-SummarizedExperiment::SummarizedExperiment(assays = list(counts=as.matrix(otu_table(x.2.0))),
                                                  colData=sample_data(x.2.0),
                                                  rowData=tax_table(x.2.0))
aux_res<-lefser(expr=se_ps,groupCol="time_point",blockCol = NULL,lda.threshold = 1)
aux_res<- merge(aux_res,tax_table(x.2.0),by.x="Names",by.y="row.names")%>% 
  dplyr::select(Names,Phylum,Class,Order,Family,Genus,scores) %>% 
  dplyr::mutate(Class=ifelse(is.na(Class),Phylum,Class)) %>% 
  dplyr::mutate(Order=ifelse(is.na(Order),Class,Order)) %>% 
  dplyr::mutate(Family=ifelse(is.na(Family),Order,Family)) %>% 
  dplyr::mutate(Genus=ifelse(is.na(Genus),paste0(Family,"_spp"),Genus)) %>% 
  dplyr::filter(! is.na(Genus)) %>% 
  dplyr::mutate(otu=Names) %>% 
  dplyr::mutate(Names=Genus) %>% 
  dplyr::select(otu,Names,scores) %>% 
  dplyr::mutate(Names = forcats::fct_reorder(Names, scores)) %>% 
  dplyr::mutate(time_point=ifelse(scores>0,"t1","t0"))
p0<-aux_res %>% 
    ggplot(.,aes(x=Names,y=scores,fill=time_point))+geom_bar(stat="identity")+coord_flip()+
    theme_classic()+ggtitle("t1 vs t0")+
    scale_fill_brewer(palette="Dark2") 
    ggsave(plot=p0,"output/Deliverables/lefser_t1_vs_t0_d_samples.pdf")
for(otu_name in aux_res$otu){
  my_ps <- x.2.0
  title <- aux_res %>% 
    dplyr::filter(otu==otu_name) 
  my_ps %>% 
  phyloseq::psmelt()%>% 
  dplyr::filter(OTU==otu_name) %>% 
  ggstatsplot::ggwithinstats(data = .,x=time_point,y=Abundance,
                                       xlab=NULL,ylab=NULL,centrality.plotting=F,
                                       bf.message=F,title=title$Names,
                                       method="non-parametric",palette="Dark2") %>% 
  
   ggsave(.,filename=paste0("output/boxplots/",title$Names,"_vs_time_point_d_Samples.pdf"),width=7, height=7)
}


### w1 samples
x.2.0 <-ps_silva %>% 
  phyloseq::subset_samples(dog_name!="Charlotte") %>% 
  phyloseq::tax_glom(.,taxrank = "Genus",NArm = F) %>%
  # phyloseq::filter_taxa(., function (x) sum(x > 0) > 2, TRUE) %>%
  phyloseq::rarefy_even_depth(.) 
x.2.0 <- x.2.0 %>% 
  phyloseq::subset_samples(treatment=="w1")
  # microbiome::transform(transform="compositional")

se_ps<-SummarizedExperiment::SummarizedExperiment(assays = list(counts=as.matrix(otu_table(x.2.0))),
                                                  colData=sample_data(x.2.0),
                                                  rowData=tax_table(x.2.0))
aux_res<-lefser(expr=se_ps,groupCol="time_point",blockCol = NULL,lda.threshold = 1)
aux_res<-merge(aux_res,tax_table(x.2.0),by.x="Names",by.y="row.names")%>% 
  dplyr::select(Names,Phylum,Class,Order,Family,Genus,scores) %>% 
  dplyr::mutate(Class=ifelse(is.na(Class),Phylum,Class)) %>% 
  dplyr::mutate(Order=ifelse(is.na(Order),Class,Order)) %>% 
  dplyr::mutate(Family=ifelse(is.na(Family),Order,Family)) %>% 
  dplyr::mutate(Genus=ifelse(is.na(Genus),paste0(Family,"_spp"),Genus)) %>% 
  dplyr::filter(! is.na(Genus)) %>% 
  dplyr::mutate(otu=Names) %>% 
  dplyr::mutate(Names=Genus) %>% 
  dplyr::select(otu,Names,scores) %>% 
  dplyr::mutate(Names = forcats::fct_reorder(Names, scores)) %>% 
  dplyr::mutate(time_point=ifelse(scores>0,"t1","t0"))
p1<-aux_res%>% 
    ggplot(.,aes(x=Names,y=scores,fill=time_point))+geom_bar(stat="identity")+coord_flip()+
    theme_classic()+ggtitle("t1 vs t0")+
    scale_fill_brewer(palette="Dark2") 
    ggsave(plot=p1,"output/Deliverables/lefser_t1_vs_t0_w1_samples.pdf")

for(otu_name in aux_res$otu){
  my_ps <- x.2.0
  title <- aux_res %>% 
    dplyr::filter(otu==otu_name) 
    
  my_ps %>% 
  phyloseq::psmelt()%>% 
  dplyr::filter(OTU==otu_name) %>% 
  ggstatsplot::ggwithinstats(data = .,x=time_point,y=Abundance,
                                       xlab=NULL,ylab=NULL,centrality.plotting=F,
                                       bf.message=F,title=title$Names,
                                       method="non-parametric",palette="Dark2") %>% 
  
   ggsave(.,filename=paste0("output/boxplots/",title$Names,"_vs_time_point_w1_Samples.pdf"),width=7, height=7)
}

### w2 samples
x.2.0 <-ps_silva %>% 
  phyloseq::subset_samples(dog_name!="Charlotte") %>% 
  phyloseq::tax_glom(.,taxrank = "Genus",NArm = F) %>%
  # phyloseq::filter_taxa(., function (x) sum(x > 0) > 2, TRUE) %>%
  phyloseq::rarefy_even_depth(.) 
x.2.0 <- x.2.0 %>% 
  phyloseq::subset_samples(treatment=="w2")
  # microbiome::transform(transform="compositional")

se_ps<-SummarizedExperiment::SummarizedExperiment(assays = list(counts=as.matrix(otu_table(x.2.0))),
                                                  colData=sample_data(x.2.0),
                                                  rowData=tax_table(x.2.0))
aux_res<-lefser(expr=se_ps,groupCol="time_point",blockCol = NULL,lda.threshold = 1)
aux_res<-merge(aux_res,tax_table(x.2.0),by.x="Names",by.y="row.names")%>% 
  dplyr::select(Names,Phylum,Class,Order,Family,Genus,scores) %>% 
  dplyr::mutate(Class=ifelse(is.na(Class),Phylum,Class)) %>% 
  dplyr::mutate(Order=ifelse(is.na(Order),Class,Order)) %>% 
  dplyr::mutate(Family=ifelse(is.na(Family),Order,Family)) %>% 
  dplyr::mutate(Genus=ifelse(is.na(Genus),paste0(Family,"_spp"),Genus)) %>% 
  dplyr::filter(! is.na(Genus)) %>% 
  dplyr::mutate(otu=Names) %>% 
  dplyr::mutate(Names=Genus) %>% 
  dplyr::select(otu,Names,scores) %>% 
  dplyr::mutate(Names = forcats::fct_reorder(Names, scores)) %>% 
  dplyr::mutate(time_point=ifelse(scores>0,"t1","t0")) 

p2<-aux_res %>% 
    ggplot(.,aes(x=Names,y=scores,fill=time_point))+geom_bar(stat="identity")+coord_flip()+
    theme_classic()+ggtitle("t1 vs t0")+
    scale_fill_brewer(palette="Dark2")
    ggsave(plot=p2,"output/Deliverables/lefser_t1_vs_t0_w2_samples.pdf")
for(otu_name in aux_res$otu){
  
  my_ps <- x.2.0
  title <- aux_res %>% 
    dplyr::filter(otu==otu_name) 
    
  my_ps %>% 
  phyloseq::psmelt()%>% 
  dplyr::filter(OTU==otu_name) %>% 
  ggstatsplot::ggwithinstats(data = .,x=time_point,y=Abundance,
                                       xlab=NULL,ylab=NULL,centrality.plotting=F,
                                       bf.message=F,title=title$Names,
                                       method="non-parametric",palette="Dark2") %>% 
   ggsave(.,filename=paste0("output/boxplots/",title$Names,"_vs_time_point_w2_Samples.pdf"),width=7, height=7)
}

p2 

```

Como ejemplo de géneros diferencialmente abundantes, se muestran las abundancias de dos de los géneros que mas diferencias presentan entre t0 y t1 para el grupo *w2*. En el caso del género *Blautia* se detecta un incremento consistente en el numero de *reads* asignados a ese género bacteriano (p\<0.07, en test apareado) para todos los animales.

```{r include graphics Blautia, eval=T, include=TRUE,echo=F, message=FALSE, warning=FALSE, paged.print=FALSE, tidy=F}
suppressMessages(require(dplyr))
suppressMessages(require(phyloseq))
suppressMessages(require(lefser))
suppressMessages(require(ggplot2))
suppressMessages(require(patchwork))

set.seed(1234)
### Genus Level
### We are really interested in comparing t1 vs t0, separately for each d,w1,w2 and d/w
title<-"Blautia"
readRDS("data/ps_silva.rds")%>% 
  phyloseq::subset_samples(dog_name!="Charlotte") %>% 
  phyloseq::tax_glom(.,taxrank = "Genus",NArm = F) %>%
  # phyloseq::filter_taxa(., function (x) sum(x > 0) > 2, TRUE) %>%
  phyloseq::rarefy_even_depth(.) %>% 
  phyloseq::subset_samples(treatment=="w2") %>% 
  phyloseq::psmelt( ) %>% 
  dplyr::filter(Genus==title) %>% 
  ggstatsplot::ggwithinstats(data = .,x=time_point,y=Abundance,
                                       xlab=NULL,ylab=NULL,centrality.plotting=F,
                                       bf.message=F,title=title,
                                       method="non-parametric",palette="Dark2")

```

En el otro sentido, el género *Faecalibacterium*, reduce su presencia o pasa a ser indetectable en *t1* en el mismo grupo de animales ( *w2* )

```{r include graphics Faecalibacterium, eval=T, include=TRUE,echo=F, message=FALSE, warning=FALSE, paged.print=FALSE, tidy=F}
suppressMessages(require(dplyr))
suppressMessages(require(phyloseq))
suppressMessages(require(lefser))
suppressMessages(require(ggplot2))
suppressMessages(require(patchwork))

set.seed(1234)
### Genus Level
### We are really interested in comparing t1 vs t0, separately for each d,w1,w2 and d/w
title<-"Faecalibacterium"
readRDS("data/ps_silva.rds")%>% 
  phyloseq::subset_samples(dog_name!="Charlotte") %>% 
  phyloseq::tax_glom(.,taxrank = "Genus",NArm = F) %>%
  # phyloseq::filter_taxa(., function (x) sum(x > 0) > 2, TRUE) %>%
  phyloseq::rarefy_even_depth(.) %>% 
  phyloseq::subset_samples(treatment=="w2") %>% 
  phyloseq::psmelt( ) %>% 
  dplyr::filter(Genus==title) %>% 
  ggstatsplot::ggwithinstats(data = .,x=time_point,y=Abundance,
                                       xlab=NULL,ylab=NULL,centrality.plotting=F,
                                       bf.message=F,title=title,
                                       method="non-parametric",palette="Dark2")

```

### Variable dry_wet

Comparativa (Wilcoxon) + análisis LDA de la abundancia de diferentes géneros bacterianos entre los grupos formados por las muestras de los perros con dieta dry(d) y wet(w)

*Deliverables: lefser_t1_vs_t0_w\_samples.pdf y boxplots independientes para cada género positivo al test*

Se observan taxas diferencialmente abundantes entre muestras del grupo de dieta seca (dry) y el grupo de muestras de dieta húmeda mediante *lefser*(<http://www.bioconductor.org/packages/release/bioc/html/lefser.html>).

-   Mayor abundancia en perros después de haber seguido dieta húmeda(w2): *Blautia, Dorea, Escherichia Shigella, grupo R.gnavus* y otros
-   Menor abundancia en perros después de haber seguido dieta húmeda(w2): *Megamonas, Prevotella, Erysipelatoclostrdium, Sutterella* y otros.

```{r DA vs time_point all dry_wet groups,eval=T, include=TRUE,echo=F, message=FALSE, warning=FALSE, paged.print=FALSE, tidy=F}

x.2.0 <-ps_silva %>% 
  phyloseq::subset_samples(dog_name!="Charlotte") %>% 
  phyloseq::tax_glom(.,taxrank = "Genus",NArm = F) %>%
  # phyloseq::filter_taxa(., function (x) sum(x > 0) > 2, TRUE) %>%
  phyloseq::rarefy_even_depth(.) %>% 
  phyloseq::subset_samples(dry_wet=="w")
  # microbiome::transform(transform="compositional")

se_ps<-SummarizedExperiment::SummarizedExperiment(assays = list(counts=as.matrix(otu_table(x.2.0))),
                                                  colData=sample_data(x.2.0),
                                                  rowData=tax_table(x.2.0))
aux_res<-lefser(expr=se_ps,groupCol="time_point",blockCol = NULL,lda.threshold = 1)
aux_res<-merge(aux_res,tax_table(x.2.0),by.x="Names",by.y="row.names")%>% 
  dplyr::select(Names,Phylum,Class,Order,Family,Genus,scores) %>% 
  dplyr::mutate(Class=ifelse(is.na(Class),Phylum,Class)) %>% 
  dplyr::mutate(Order=ifelse(is.na(Order),Class,Order)) %>% 
  dplyr::mutate(Family=ifelse(is.na(Family),Order,Family)) %>% 
  dplyr::mutate(Genus=ifelse(is.na(Genus),paste0(Family,"_spp"),Genus)) %>% 
  dplyr::filter(! is.na(Genus)) %>% 
  dplyr::mutate(otu=Names) %>% 
  dplyr::mutate(Names=Genus) %>% 
  dplyr::select(otu,Names,scores) %>% 
  dplyr::mutate(Names = forcats::fct_reorder(Names, scores)) %>% 
  dplyr::mutate(time_point=ifelse(scores>0,"t1","t0")) 

p0<- aux_res %>% 
    ggplot(.,aes(x=Names,y=scores,fill=time_point))+geom_bar(stat="identity")+coord_flip()+
    theme_classic()+ggtitle("t1 vs t0")+
    scale_fill_brewer(palette="Dark2")
ggsave(plot=p0,"output/Deliverables/lefser_t1_vs_t0_w_samples.pdf")

for(otu_name in aux_res$otu){
  my_ps <- x.2.0
  title <- aux_res %>% 
    dplyr::filter(otu==otu_name) 
    
  my_ps %>% 
  phyloseq::psmelt()%>% 
  dplyr::filter(OTU==otu_name) %>% 
  ggstatsplot::ggwithinstats(data = .,x=time_point,y=Abundance,
                                       xlab=NULL,ylab=NULL,centrality.plotting=F,
                                       bf.message=F,title=title$Names,
                                       method="non-parametric",palette="Dark2") %>% 
  
   ggsave(.,filename=paste0("output/boxplots/",title$Names,"_vs_time_point_wet_Samples.pdf"),width=7, height=7)
}

```

### Ordenación estratificada por grupos de *treatment*

Usaremos análisis de ordenación para visualizar los cambios en la estructura del microbioma, independientemente, en cada grupo definido por la variable principal *treatment*.

*Deliverables: Ordination_by_treatment_by_time_point.pdf*

Se puede observar un cambio consistente en todos los animales que forman el grupo *w2* en el espacio de ordenación, ligado a la proporción de variabilidad explicada en el análisis PERMANOVA anterior. De la misma manera, se puede observar que los cambios en los grupos de *treatment* restantes no son consistentes y no producen un *clustering* en el espacio de ordenación.

```{r ordination by treatment, include=TRUE,echo=F, message=FALSE, warning=FALSE, paged.print=FALSE, tidy=F}
suppressWarnings(require(phyloseq))
suppressWarnings(require(phylosmith))
suppressWarnings(require(patchwork))
suppressWarnings(require(vegan))
suppressWarnings(require(ggplot2))
suppressWarnings(require(ggforce))
x.2.0<-readRDS("data/ps_silva.rds")


#### treatment == "d"
set.seed(1234)
x.2.0.genus<-tax_glom(x.2.0,taxrank="Genus",NArm = F) %>% 
  microbiome::transform(.,transform="compositional") %>% 
  phyloseq::subset_samples(.,treatment=="d")

sampleData<-data.frame(sample_data(x.2.0.genus))
rownames(sampleData)<-sampleData$SampleID
sample_data(x.2.0.genus)<-sampleData
capture.output(myOrd<-ordinate(x.2.0.genus,method="NMDS",distance="wunifrac"),file = "/dev/null")

p.4.0.samples=plot_ordination(x.2.0.genus,myOrd)
p0<-p.4.0.samples  + geom_point(aes(color=time_point)) +ggtitle("NMDS(WUnifrac) by time_point / d samples")+
  theme(panel.border=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.line=element_line(colour="black"),axis.text.x=element_text(size=16),axis.text.y=element_text(size=16))+
  theme(axis.title.x=element_text(size=14),axis.title.y=element_text(size=14))+
  theme(legend.text=element_text(size=12,face="italic"),legend.title=element_text(size=12))+
  theme(plot.title=element_text(lineheight=1,face="bold",size=15))+stat_ellipse(alpha=1,mapping=aes(color=time_point))+
  theme_classic() + theme(axis.line.x = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.line.y = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.text.x = element_text(size = 10), 
                                    axis.text.y = element_text(size = 10),
                                    axis.title.x = element_text(size = 10, 
                                        face = "bold"), axis.title.y = element_text(size = 10, 
                                        face = "bold"), legend.title = element_text(size = 10, 
                                        face = "bold"), legend.text = element_text(size = 8), 
                                        legend.spacing.x = unit(0.005, "npc"), 
                                        legend.key.size = unit(4,"mm"), legend.background = element_rect(fill = (alpha = 0))) + 
        labs(x = "NMDS Dimension 1", y = "NMDS Dimension 2") + geom_line(aes(group=dog_name),color="grey")+
      theme(plot.title = element_text(size = 10, face = "bold"))+
        guides(fill = guide_legend(override.aes = list(size = 4)))


#### treatment == "w1"
set.seed(1234)
x.2.0.genus<-tax_glom(x.2.0,taxrank="Genus",NArm = F) %>% 
  microbiome::transform(.,transform="compositional") %>% 
  phyloseq::subset_samples(.,treatment=="w1")

sampleData<-data.frame(sample_data(x.2.0.genus))
rownames(sampleData)<-sampleData$SampleID
sample_data(x.2.0.genus)<-sampleData
capture.output(myOrd<-ordinate(x.2.0.genus,method="NMDS",distance="wunifrac"),file = "/dev/null")

p.4.0.samples=plot_ordination(x.2.0.genus,myOrd)
p1<-p.4.0.samples  + geom_point(aes(color=time_point)) +ggtitle("NMDS(WUnifrac) by time_point / w1 samples")+
  theme(panel.border=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.line=element_line(colour="black"),axis.text.x=element_text(8),axis.text.y=element_text(size=8))+
  theme(axis.title.x=element_text(size=14),axis.title.y=element_text(size=14))+
  theme(legend.text=element_text(size=12,face="italic"),legend.title=element_text(size=12))+
  theme(plot.title=element_text(lineheight=1,face="bold",size=15))+stat_ellipse(alpha=1,mapping=aes(color=time_point))+
  theme_classic() + theme(axis.line.x = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.line.y = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.text.x = element_text(size = 10), 
                                    axis.text.y = element_text(size = 10),
                                    axis.title.x = element_text(size = 10, 
                                        face = "bold"), axis.title.y = element_text(size = 10, 
                                        face = "bold"), legend.title = element_text(size = 10, 
                                        face = "bold"), legend.text = element_text(size = 8), 
                                        legend.spacing.x = unit(0.005, "npc"), 
                                        legend.key.size = unit(4,"mm"), legend.background = element_rect(fill = (alpha = 0))) + 
        labs(x = "NMDS Dimension 1", y = "NMDS Dimension 2") + geom_line(aes(group=dog_name),color="grey")+
        theme(plot.title = element_text(size = 10, face = "bold"))+
        guides(fill = guide_legend(override.aes = list(size = 4)))


#### treatment == "w2"
set.seed(1234)
x.2.0.genus<-tax_glom(x.2.0,taxrank="Genus",NArm = F) %>% 
  microbiome::transform(.,transform="compositional") %>% 
  phyloseq::subset_samples(.,treatment=="w2")

sampleData<-data.frame(sample_data(x.2.0.genus))
rownames(sampleData)<-sampleData$SampleID
sample_data(x.2.0.genus)<-sampleData
capture.output(myOrd<-ordinate(x.2.0.genus,method="NMDS",distance="wunifrac"),file = "/dev/null")

p.4.0.samples=plot_ordination(x.2.0.genus,myOrd)
p2<-p.4.0.samples  + geom_point(aes(color=time_point)) +ggtitle("NMDS(WUnifrac) by time_point / w2 samples")+
  theme(panel.border=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.line=element_line(colour="black"),axis.text.x=element_text(size=16),axis.text.y=element_text(size=16))+
  theme(axis.title.x=element_text(size=14),axis.title.y=element_text(size=14))+
  theme(legend.text=element_text(size=12,face="italic"),legend.title=element_text(size=12))+
  theme(plot.title=element_text(lineheight=1,face="bold",size=15))+stat_ellipse(alpha=1,mapping=aes(color=time_point))+
  theme_classic() + theme(axis.line.x = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.line.y = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.text.x = element_text(size = 10), 
                                    axis.text.y = element_text(size = 10),
                                    axis.title.x = element_text(size = 10, 
                                        face = "bold"), axis.title.y = element_text(size = 10, 
                                        face = "bold"), legend.title = element_text(size = 10, 
                                        face = "bold"), legend.text = element_text(size = 8), 
                                        legend.spacing.x = unit(0.005, "npc"), 
                                        legend.key.size = unit(4,"mm"), legend.background = element_rect(fill = (alpha = 0))) + 
        labs(x = "NMDS Dimension 1", y = "NMDS Dimension 2")+ geom_line(aes(group=dog_name),color="grey")+
        theme(plot.title = element_text(size = 10, face = "bold"))+
        guides(fill = guide_legend(override.aes = list(size = 4)))

#### time_point == "t1"
set.seed(1234)
x.2.0.genus<-tax_glom(x.2.0,taxrank="Genus",NArm = F) %>% 
  microbiome::transform(.,transform="compositional") %>% 
  phyloseq::subset_samples(.,time_point=="t1")

sampleData<-data.frame(sample_data(x.2.0.genus))
rownames(sampleData)<-sampleData$SampleID
sample_data(x.2.0.genus)<-sampleData
capture.output(myOrd<-ordinate(x.2.0.genus,method="NMDS",distance="wunifrac"),file = "/dev/null")

p.4.0.samples=plot_ordination(x.2.0.genus,myOrd)
p3<-p.4.0.samples  + geom_point(aes(color=treatment)) +ggtitle("NMDS(WUnifrac) by treatment (t1 samples)")+
  theme(panel.border=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.line=element_line(colour="black"),axis.text.x=element_text(size=16),axis.text.y=element_text(size=16))+
  theme(axis.title.x=element_text(size=14),axis.title.y=element_text(size=14))+
  theme(legend.text=element_text(size=12,face="italic"),legend.title=element_text(size=12))+
  theme(plot.title=element_text(lineheight=1,face="bold",size=15))+stat_ellipse(alpha=1,mapping=aes(color=treatment))+
  theme_classic() + theme(axis.line.x = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.line.y = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.text.x = element_text(size = 10), 
                                    axis.text.y = element_text(size = 10),
                                    axis.title.x = element_text(size = 10, 
                                        face = "bold"), axis.title.y = element_text(size = 10, 
                                        face = "bold"), legend.title = element_text(size = 10, 
                                        face = "bold"), legend.text = element_text(size = 8), 
                                        legend.spacing.x = unit(0.005, "npc"), 
                                        legend.key.size = unit(4,"mm"), legend.background = element_rect(fill = (alpha = 0)))+
        labs(x = "NMDS Dimension 1", y = "NMDS Dimension 2") +theme(plot.title = element_text(size = 10, face = "bold"))+
        guides(fill = guide_legend(override.aes = list(size = 4)))

pT <- (p0 | p1 ) / (p2  | p3)
pT
ggsave(pT,filename = "output/Deliverables/Ordination_by_treatment_by_time_point.pdf",width=15,height=15)
```
